{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b5faf083-ac8b-47db-8ad8-763dcb6b7bcd",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# üíñFunüíñGutüíñ"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cdf3ae77-e2f5-4021-8d66-64132946272b",
   "metadata": {},
   "source": [
    "Hi! Welcome to FunGut, the ETH spinoff determined to develop a commercially available platform for users to analyze their gut mycobiome, i.e. fungal microbiome, in order to gain personalized health insights. We recently recieved fecal samples from people living in Europe, Noth Amrtica and Oceania for a pilot trial with the aim of finding relevant factors that affect the gut mycobiome. In this notebook, we will analyze the DNA features that were sequenced from our samples to check if we find any variables that significantly affect either the diversities (both within and between samples) or the differential abundances of features in our samples. Let's begin!"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20b5d4a5-5640-4acd-8c97-7f7555307126",
   "metadata": {
    "tags": []
   },
   "source": [
    "TODO:\n",
    "    \n",
    "- Diversity Analsysis nomau outputs aluege und bestimme was signifikant\n",
    "- Differential analysis d√∂rt mache wo in div. ana. relevant\n",
    "\n",
    "- Denn das beides ou bim kmerizer mache\n",
    "- Denn vrgliche"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "19941a9b-a21b-4edb-aab3-311eca5dabb5",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 00 Packages and Directory üì¶"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70ea3ee1-b95a-4429-8609-6cb63c14b71f",
   "metadata": {},
   "source": [
    "Before we do anything, we need to make sure we have all the tools we need to do analyse our data and that we have a system put in place to organize our inputs and outputs. Here we first install geopandas which we use to look at our metadata later on. Subsequently we import all required plugins, set our data directory and create a simple numbered directory to organize our outputs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "745019d1-f0f9-4343-b560-fd440864512f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: geopandas in /opt/conda/lib/python3.10/site-packages (1.1.1)\n",
      "Requirement already satisfied: numpy>=1.24 in /opt/conda/lib/python3.10/site-packages (from geopandas) (1.26.4)\n",
      "Requirement already satisfied: pyogrio>=0.7.2 in /opt/conda/lib/python3.10/site-packages (from geopandas) (0.12.1)\n",
      "Requirement already satisfied: packaging in /opt/conda/lib/python3.10/site-packages (from geopandas) (25.0)\n",
      "Requirement already satisfied: pandas>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from geopandas) (2.2.2)\n",
      "Requirement already satisfied: pyproj>=3.5.0 in /opt/conda/lib/python3.10/site-packages (from geopandas) (3.7.1)\n",
      "Requirement already satisfied: shapely>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from geopandas) (2.1.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in /opt/conda/lib/python3.10/site-packages (from pandas>=2.0.0->geopandas) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2020.1 in /opt/conda/lib/python3.10/site-packages (from pandas>=2.0.0->geopandas) (2025.2)\n",
      "Requirement already satisfied: tzdata>=2022.7 in /opt/conda/lib/python3.10/site-packages (from pandas>=2.0.0->geopandas) (2025.2)\n",
      "Requirement already satisfied: certifi in /opt/conda/lib/python3.10/site-packages (from pyogrio>=0.7.2->geopandas) (2025.7.14)\n",
      "Requirement already satisfied: six>=1.5 in /opt/conda/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas>=2.0.0->geopandas) (1.17.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install geopandas "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "968e51e9-6730-40b0-9e95-b4a2d4c3888e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import geopandas as gpd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "56a5745e-add5-4f1e-907d-c95b249b4e45",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import IPython\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np  \n",
    "\n",
    "import qiime2 as q2\n",
    "from qiime2 import Visualization\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a208757-7060-4a2c-be0a-cde75dec939b",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 00.01 File directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bd4f09a7-d5cd-48ce-bb45-43e6f02e4b85",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "! mkdir -p \"data/01\"\n",
    "! mkdir -p \"data/02\"\n",
    "! mkdir -p \"data/03\"\n",
    "! mkdir -p \"data/03/denoising_4\"\n",
    "! mkdir -p \"data/04\"\n",
    "! mkdir -p \"data/05\"\n",
    "! mkdir -p \"data/06\"\n",
    "! mkdir -p \"data/06/boots-kmer-diversity\"\n",
    "! mkdir -p \"data/06/alpha_diversity\"\n",
    "! mkdir -p \"data/06/beta_diversity_plots\"\n",
    "! mkdir -p \"data/06/permanova\"\n",
    "! mkdir -p \"data/07\"\n",
    "! mkdir -p \"data/07/ancombc\"\n",
    "! mkdir -p \"data/08\"\n",
    "! mkdir -p \"data/08/alpha_diversity\"\n",
    "! mkdir -p \"data/08/beta_diversity_plots\"\n",
    "! mkdir -p \"data/08/permanova\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c99db570-c6a5-48fd-a570-d48b4f44e23f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "data_dir = 'data'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38445ca3-9c18-4c47-8252-26fe43e9cd6c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 01 Data import üì•"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "482a9a0b-d56f-4556-b2a0-74c80cc5033a",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 01.01 Importing raw data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "90be2135-1a36-4b5f-8d3b-c00fb9ef60e3",
   "metadata": {},
   "source": [
    "Having set up our working environment, lets get started with importing the raw metadata and the raw sequenes. After importing, we cna perform first checks to see what our sequences look like by using qiime tools peek and creating a demux feature table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4cd0d918-4508-44b5-88b3-dac09616a4e4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 18:53:26--  https://polybox.ethz.ch/index.php/s/bLQ6eRWEZo8KmFo/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 18798 (18K) [application/octet-stream]\n",
      "Saving to: ‚Äòdata/fungut_metadata.tsv‚Äô\n",
      "\n",
      "data/fungut_metadat 100%[===================>]  18.36K  --.-KB/s    in 0.008s  \n",
      "\n",
      "2025-12-02 18:53:26 (2.30 MB/s) - ‚Äòdata/fungut_metadata.tsv‚Äô saved [18798/18798]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/bLQ6eRWEZo8KmFo/download\" -O $data_dir/fungut_metadata.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ab1e1f07-9a47-4b8a-9325-9100446dd29c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 18:53:27--  https://polybox.ethz.ch/index.php/s/fe7AYe2fBR9jaab/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 712595535 (680M) [application/octet-stream]\n",
      "Saving to: ‚Äòdata/fungut_forward_reads.qza‚Äô\n",
      "\n",
      "data/fungut_forward 100%[===================>] 679.58M   398MB/s    in 1.7s    \n",
      "\n",
      "2025-12-02 18:53:28 (398 MB/s) - ‚Äòdata/fungut_forward_reads.qza‚Äô saved [712595535/712595535]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/fe7AYe2fBR9jaab/download\" -O data/fungut_forward_reads.qza "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "66d50052-d11c-4597-86cd-16a60c8e982e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mUUID\u001b[0m:        3638611d-1767-413b-9390-70ee3d78e4ff\n",
      "\u001b[32mType\u001b[0m:        SampleData[SequencesWithQuality]\n",
      "\u001b[32mData format\u001b[0m: SingleLanePerSampleSingleEndFastqDirFmt\n"
     ]
    }
   ],
   "source": [
    "!qiime tools peek \"$data_dir/fungut_forward_reads.qza\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a279d38a-b3ee-4e2b-a12d-6b2153ddeb36",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/01/demux_summary.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime demux summarize \\\n",
    "    --i-data $data_dir/fungut_forward_reads.qza \\\n",
    "    --o-visualization $data_dir/01/demux_summary.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c86a0e8d-b04c-4679-97ad-9c72eaf093e3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/f35a848d-ec05-4549-ad4b-c4e1a8a83293')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f35a848d-ec05-4549-ad4b-c4e1a8a83293>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/01/demux_summary.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "87c8f90c-fed2-4ce8-bf49-3db0c4c0c4bd",
   "metadata": {
    "tags": []
   },
   "source": [
    "<div style=\"background-color: lightblue; padding: 20px;\">\n",
    "Analysing our demultiplexed sequences using the demux feature table, we gather the following information:\n",
    "</div>\n",
    "\n",
    "- Our sequences contain a minimum of 20567 foreward reads (FR) per sample\n",
    "- The mean number of FR is 64221.42 \n",
    "- The maximum of FR is 113461\n",
    "- The total number of FR is 9633213\n",
    "- Quality score for the FR is stable for the entire sequencing length with the median quality score for position 151 (maximum length) being 38 and few medians dipping only slightly below that.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2f6ed1f3-e61e-461a-b476-c8903cf24b41",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 01.02 Metadata editing"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b6ccab30-cee4-4592-92ff-815278d657cd",
   "metadata": {},
   "source": [
    "Having downloaded our metadata file in the previous chapter, we will now need to edit it slightly to ensure we don't run into any errors in the downstream processing. First, we will replace all numeric values that currently are strings of \"Not provided\" into Nans so we can work with these numbers even though we don't have a set numeric value. For the categorical variables, we want to keep \"Not provided\" so that we don't loose them in our diversity analysis and differential abundance analysis. Once we have cleaned the \"Not provided\" strings, we will upload the new metadata file.\n",
    "\n",
    "In a second step after cleaning the file, we want to add a column for the rural/urban score of each sample based on the coordinates that are already present in the metadata. We do this by downloading our cleaned file, using QGIS and the RUCC (Rural Urban Conrinuum Codes) and import the edited file with the new column that includes the rural/urban score.\n",
    "\n",
    "Once all that is done, we will also add a new column to group ranges of BMI to their corresponding category on the BMI scale as a last modification of our metadata before we can get to work.\n",
    "\n",
    "And with that, we have our finalised metadata file (for now...)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "cabbdffb-2655-4298-b743-6a19afda768f",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_clean = pd.read_csv(f\"{data_dir}/fungut_metadata.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "1c4e786a-6a27-42a5-949b-f617a873c883",
   "metadata": {},
   "outputs": [],
   "source": [
    "numeric_cols = [\"latitude_sample\", \"longitude_sample\", \"age_years_sample\", \"height_cm_sample\", \"weight_kg_sample\", \"bmi_sample\",] \n",
    "\n",
    "for col in numeric_cols:\n",
    "    if col in metadata_clean.columns:\n",
    "        metadata_clean[col] = metadata_clean[col].replace(\n",
    "            [\"Not provided\"],\n",
    "            np.nan\n",
    "        )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "49c99a3d-6c0d-46ef-8866-2f1b0c2fd6cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_clean.to_csv(\"data/metadata_clean.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35c93add-fcc4-4294-aaee-530c3b1660e8",
   "metadata": {},
   "source": [
    "#### Rural Urban Continuum Codes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "695e0351-8699-49a8-b72c-c3ca7847d0e4",
   "metadata": {},
   "source": [
    "We can now use this cleaned metadata file to add in additional columns to enable us to look at more variables in downstream processing. Firstly, we are interested in not only the geographical coordinates where our samples were taken (already provided) but also in wether or not these locations are rural or urban. As diets and other environmental factors can strongly affect the gut microbiome of our sample hosts, we can later examine if we see any difference between the following three categories: rural, peri-urban and urban.\n",
    "\n",
    "We add this column using the software QGIS and we apply the RUCC (Rural Urban Continuum Codes) system to attribute one of those three categories to our samples. This step was performed in QGIS directly and thus we need to import the newly modified metadata file again (we needed the NaNs for QGIS, this is why we had to use this order of metadata editing)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "36b7cbab-ff43-4371-a41e-dc783ca3b2de",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 18:54:57--  https://polybox.ethz.ch/index.php/s/7XXTnp4in3rYqJQ/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 20576 (20K) [application/octet-stream]\n",
      "Saving to: ‚Äòdata/metadata_rucc.tsv‚Äô\n",
      "\n",
      "data/metadata_rucc. 100%[===================>]  20.09K  --.-KB/s    in 0s      \n",
      "\n",
      "2025-12-02 18:54:57 (43.7 MB/s) - ‚Äòdata/metadata_rucc.tsv‚Äô saved [20576/20576]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/7XXTnp4in3rYqJQ/download\" -O $data_dir/metadata_rucc.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "b838a6d2-9b03-49a2-b052-ee25e43dc875",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_rucc = pd.read_csv(f\"{data_dir}/metadata_rucc.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c2ad8183-e969-4d5d-95ac-d7d3ec2dbc3d",
   "metadata": {},
   "source": [
    "#### BMI Groups"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa3e4f87-4e1c-4e0c-ba22-1b4a7969c731",
   "metadata": {},
   "source": [
    "We are further interested in the correlation of BMI with gut microbiome. To better analyze this later on in the pipeline, we create groups of BMI scores based on the official groupings of the National Institute of Health of the USA (most of our samples are from the USA). We do this by creating bins and labels and then applying these to our metadata file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2fbd2dd8-70f6-41d1-b87a-11907d4f08b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "bins_bmi = [0, 18.5, 24.9, 29.9, 34.9, 39.9, float('inf')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "44f4c9f4-8c28-403d-87bc-5be7a4d07594",
   "metadata": {},
   "outputs": [],
   "source": [
    "labels_bmi = ['Underweight', 'Normal', 'Overweight', 'Adipose 1', 'Adipose 2', 'Adipose 3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "573d05e1-4878-4846-94bf-ae3be0b9ec5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_rucc['BMI_category'] = pd.cut(\n",
    "    metadata_rucc['bmi_sample'],\n",
    "    bins=bins_bmi,\n",
    "    labels=labels_bmi,\n",
    "    right=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "63a7b9a9-53d5-46fe-b08a-0958e9f6426e",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_rucc.to_csv(\"data/metadata_ed.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c1e606e-f5dd-42ca-ac28-2a10da295087",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 01.03 Metadata exploration"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db9f6645-9b4c-48ea-8294-6a2f95740d6f",
   "metadata": {},
   "source": [
    "Having freshly cleaned and modified our metadata file, let's have a look at what is actually inside and find out more about where our samples come from."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "360f1420-0cf2-47bd-bd56-9fc078d85249",
   "metadata": {},
   "outputs": [],
   "source": [
    "#graphe vo livia"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee3bd590-4f5d-4d00-a390-88fc4f99a40b",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 02 Trimming primers ‚úÇÔ∏è"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be002cae-0a13-47ad-b3c2-e12be3d1b826",
   "metadata": {
    "tags": []
   },
   "source": [
    "In this chapter we remove the primers from the sequences and then create a new summary table to check if we see any impact on the sequence length distribution (Spoiler alert: We don't)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f0169870-a2d3-45f7-b6e6-d72efdb4b478",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327198_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327198_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327198_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327198_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        35,631 reads @   4.5 ¬µs/read;  13.26 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  35,631\n",
      "Reads with adapters:                        56 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        35,631 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,350,207 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,350,000 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 56 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t556.7\t0\t18\n",
      "4\t37\t139.2\t0\t37\n",
      "5\t1\t34.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327199_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327199_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327199_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327199_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,532 reads @   4.5 ¬µs/read;  13.24 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,532\n",
      "Reads with adapters:                        69 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,532 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,431,308 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,431,040 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 69 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t1086.4\t0\t16\n",
      "4\t48\t271.6\t0\t48\n",
      "5\t2\t67.9\t0\t2\n",
      "6\t3\t17.0\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327266_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327266_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327266_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327266_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        55,560 reads @   4.3 ¬µs/read;  13.91 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  55,560\n",
      "Reads with adapters:                        54 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        55,560 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,387,886 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,387,677 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 54 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t868.1\t0\t11\n",
      "4\t41\t217.0\t0\t41\n",
      "6\t2\t13.6\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327282_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327282_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327282_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327282_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        77,651 reads @   4.3 ¬µs/read;  13.97 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  77,651\n",
      "Reads with adapters:                        64 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        77,651 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,682,432 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,682,192 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 64 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t1213.3\t0\t17\n",
      "4\t46\t303.3\t0\t46\n",
      "5\t1\t75.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327284_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327284_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327284_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327284_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        20,567 reads @   5.0 ¬µs/read;  11.98 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  20,567\n",
      "Reads with adapters:                        21 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        20,567 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,103,965 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      3,103,888 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 21 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t321.4\t0\t7\n",
      "4\t14\t80.3\t0\t14\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327285_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327285_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327285_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327285_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        72,983 reads @   4.3 ¬µs/read;  13.86 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  72,983\n",
      "Reads with adapters:                        92 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        72,983 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,016,656 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,016,307 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 92 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t21\t1140.4\t0\t21\n",
      "4\t70\t285.1\t0\t70\n",
      "6\t1\t17.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327287_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327287_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327287_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327287_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        51,395 reads @   4.5 ¬µs/read;  13.21 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  51,395\n",
      "Reads with adapters:                        49 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        51,395 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,758,649 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,758,463 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 49 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t803.0\t0\t10\n",
      "4\t39\t200.8\t0\t39\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327288_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327288_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327288_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327288_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        79,528 reads @   4.7 ¬µs/read;  12.71 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  79,528\n",
      "Reads with adapters:                        64 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        79,528 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,930,662 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,930,421 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 64 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t1242.6\t0\t17\n",
      "4\t45\t310.7\t0\t45\n",
      "5\t2\t77.7\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327289_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327289_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327289_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327289_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        81,068 reads @   4.5 ¬µs/read;  13.34 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  81,068\n",
      "Reads with adapters:                        79 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        81,068 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,239,266 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,238,954 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t1266.7\t0\t13\n",
      "4\t64\t316.7\t0\t64\n",
      "5\t1\t79.2\t0\t1\n",
      "12\t1\t0.0\t1\t0 1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327300_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327300_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327300_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327300_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        60,148 reads @   5.2 ¬µs/read;  11.47 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  60,148\n",
      "Reads with adapters:                        62 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        60,148 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,078,963 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,078,734 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 62 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t939.8\t0\t19\n",
      "4\t43\t235.0\t0\t43\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327303_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327303_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327303_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327303_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        39,873 reads @   4.3 ¬µs/read;  13.80 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  39,873\n",
      "Reads with adapters:                        27 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        39,873 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,020,278 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,020,179 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 27 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9\t623.0\t0\t9\n",
      "4\t18\t155.8\t0\t18\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327305_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327305_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327305_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327305_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,543 reads @   4.6 ¬µs/read;  13.14 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,543\n",
      "Reads with adapters:                        49 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,543 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,439,292 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,439,105 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 49 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12\t1086.6\t0\t12\n",
      "4\t34\t271.7\t0\t34\n",
      "5\t3\t67.9\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327308_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327308_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327308_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327308_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        90,116 reads @   4.3 ¬µs/read;  13.85 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  90,116\n",
      "Reads with adapters:                        78 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        90,116 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,605,312 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,605,022 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t25\t1408.1\t0\t25\n",
      "4\t50\t352.0\t0\t50\n",
      "5\t3\t88.0\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327311_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327311_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327311_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327311_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        62,730 reads @   4.3 ¬µs/read;  14.02 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  62,730\n",
      "Reads with adapters:                        78 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        62,730 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,470,391 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,470,104 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t26\t980.2\t0\t26\n",
      "4\t51\t245.0\t0\t51\n",
      "5\t1\t61.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327313_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327313_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327313_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327313_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        61,432 reads @   4.3 ¬µs/read;  14.03 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  61,432\n",
      "Reads with adapters:                        75 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        61,432 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,220,671 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,220,388 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 75 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t959.9\t0\t19\n",
      "4\t55\t240.0\t0\t55\n",
      "6\t1\t15.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327314_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327314_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327314_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327314_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        54,500 reads @   4.3 ¬µs/read;  13.82 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  54,500\n",
      "Reads with adapters:                        45 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        54,500 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,216,138 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,215,973 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 45 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t851.6\t0\t15\n",
      "4\t30\t212.9\t0\t30\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327316_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327316_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327316_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327316_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        77,716 reads @   4.3 ¬µs/read;  14.08 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  77,716\n",
      "Reads with adapters:                       101 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        77,716 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,724,726 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,724,343 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t21\t1214.3\t0\t21\n",
      "4\t80\t303.6\t0\t80\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327317_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327317_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327317_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327317_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        56,804 reads @   4.5 ¬µs/read;  13.26 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  56,804\n",
      "Reads with adapters:                        60 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        56,804 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,571,877 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,571,650 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 60 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t887.6\t0\t14\n",
      "4\t45\t221.9\t0\t45\n",
      "5\t1\t55.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327318_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327318_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327318_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327318_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,301 reads @   4.5 ¬µs/read;  13.45 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,301\n",
      "Reads with adapters:                        80 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,301 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,369,216 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,368,898 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 80 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t1176.6\t0\t14\n",
      "4\t62\t294.1\t0\t62\n",
      "5\t1\t73.5\t0\t1\n",
      "6\t2\t18.4\t0\t2\n",
      "11\t1\t0.0\t1\t0 1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327322_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327322_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327322_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327322_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,741 reads @   4.4 ¬µs/read;  13.76 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,741\n",
      "Reads with adapters:                       101 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,741 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,464,962 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,464,581 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t33\t1089.7\t0\t33\n",
      "4\t61\t272.4\t0\t61\n",
      "5\t5\t68.1\t0\t5\n",
      "6\t1\t17.0\t0\t1\n",
      "7\t1\t4.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327323_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327323_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327323_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327323_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        68,793 reads @   4.3 ¬µs/read;  13.86 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,793\n",
      "Reads with adapters:                        44 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,793 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,368,129 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,367,961 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 44 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t1074.9\t0\t10\n",
      "4\t32\t268.7\t0\t32\n",
      "5\t2\t67.2\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327325_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327325_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327325_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327325_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        32,700 reads @   4.5 ¬µs/read;  13.41 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  32,700\n",
      "Reads with adapters:                        35 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        32,700 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,934,245 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,934,110 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 35 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t510.9\t0\t5\n",
      "4\t30\t127.7\t0\t30\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327326_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327326_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327326_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327326_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        81,166 reads @   4.3 ¬µs/read;  13.88 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  81,166\n",
      "Reads with adapters:                        59 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        81,166 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,255,190 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,254,972 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 59 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1268.2\t0\t20\n",
      "4\t38\t317.1\t0\t38\n",
      "6\t1\t19.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327327_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327327_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327327_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327327_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        84,879 reads @   4.2 ¬µs/read;  14.23 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  84,879\n",
      "Reads with adapters:                        79 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        84,879 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,814,733 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,814,445 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t1326.2\t0\t29\n",
      "4\t49\t331.6\t0\t49\n",
      "5\t1\t82.9\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327329_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327329_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327329_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327329_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        54,969 reads @   4.9 ¬µs/read;  12.32 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  54,969\n",
      "Reads with adapters:                        34 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        54,969 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,287,105 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,286,979 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 34 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t858.9\t0\t11\n",
      "4\t22\t214.7\t0\t22\n",
      "5\t1\t53.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327332_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327332_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327332_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327332_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        66,003 reads @   4.5 ¬µs/read;  13.43 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  66,003\n",
      "Reads with adapters:                        63 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        66,003 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,965,547 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,965,315 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 63 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t21\t1031.3\t0\t21\n",
      "4\t41\t257.8\t0\t41\n",
      "5\t1\t64.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327335_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327335_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327335_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327335_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        62,732 reads @   4.6 ¬µs/read;  13.06 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  62,732\n",
      "Reads with adapters:                        67 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        62,732 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,447,310 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,447,056 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 67 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t980.2\t0\t17\n",
      "4\t48\t245.0\t0\t48\n",
      "5\t1\t61.3\t0\t1\n",
      "6\t1\t15.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327338_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327338_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327338_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327338_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        71,603 reads @   4.3 ¬µs/read;  14.07 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  71,603\n",
      "Reads with adapters:                        74 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        71,603 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,745,232 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,744,939 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 74 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9\t1118.8\t0\t9\n",
      "4\t62\t279.7\t0\t62\n",
      "6\t3\t17.5\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327340_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327340_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327340_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327340_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        35,313 reads @   4.3 ¬µs/read;  13.84 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  35,313\n",
      "Reads with adapters:                        30 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        35,313 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,299,801 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,299,690 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 30 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t551.8\t0\t10\n",
      "4\t19\t137.9\t0\t19\n",
      "5\t1\t34.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327343_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327343_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327343_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327343_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        85,486 reads @   4.6 ¬µs/read;  13.04 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  85,486\n",
      "Reads with adapters:                       104 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        85,486 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,905,958 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,905,574 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 104 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t32\t1335.7\t0\t32\n",
      "4\t72\t333.9\t0\t72\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327344_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327344_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327344_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327344_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        55,454 reads @   4.3 ¬µs/read;  13.89 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  55,454\n",
      "Reads with adapters:                        67 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        55,454 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,368,533 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,368,278 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 67 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t866.5\t0\t15\n",
      "4\t50\t216.6\t0\t50\n",
      "5\t2\t54.2\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327346_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327346_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327346_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327346_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        85,051 reads @   4.2 ¬µs/read;  14.43 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  85,051\n",
      "Reads with adapters:                        81 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        85,051 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,813,424 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,813,124 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 81 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t24\t1328.9\t0\t24\n",
      "4\t57\t332.2\t0\t57\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327348_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327348_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327348_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327348_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        86,511 reads @   4.4 ¬µs/read;  13.71 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  86,511\n",
      "Reads with adapters:                        78 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        86,511 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,060,814 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,060,528 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1351.7\t0\t27\n",
      "4\t50\t337.9\t0\t50\n",
      "5\t1\t84.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327349_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327349_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327349_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327349_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        83,668 reads @   4.4 ¬µs/read;  13.63 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  83,668\n",
      "Reads with adapters:                        80 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        83,668 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,631,243 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,630,934 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 80 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t1307.3\t0\t15\n",
      "4\t63\t326.8\t0\t63\n",
      "6\t2\t20.4\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327351_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327351_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327351_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327351_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        57,926 reads @   4.3 ¬µs/read;  14.07 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  57,926\n",
      "Reads with adapters:                        48 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        57,926 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,691,477 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,691,292 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 48 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t905.1\t0\t10\n",
      "4\t36\t226.3\t0\t36\n",
      "5\t1\t56.6\t0\t1\n",
      "6\t1\t14.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327352_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327352_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327352_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327352_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        32,418 reads @   4.4 ¬µs/read;  13.55 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  32,418\n",
      "Reads with adapters:                        60 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        32,418 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,891,586 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,891,369 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 60 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t28\t506.5\t0\t28\n",
      "4\t29\t126.6\t0\t29\n",
      "5\t1\t31.7\t0\t1\n",
      "6\t2\t7.9\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327353_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327353_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327353_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327353_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        61,291 reads @   4.3 ¬µs/read;  13.91 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  61,291\n",
      "Reads with adapters:                        57 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        61,291 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,234,220 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,234,001 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 57 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9\t957.7\t0\t9\n",
      "4\t48\t239.4\t0\t48\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327354_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327354_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327354_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327354_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        59,999 reads @   4.3 ¬µs/read;  13.96 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  59,999\n",
      "Reads with adapters:                        49 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        59,999 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,056,215 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,056,029 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 49 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t937.5\t0\t10\n",
      "4\t39\t234.4\t0\t39\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327355_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327355_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327355_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327355_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        60,138 reads @   4.2 ¬µs/read;  14.43 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  60,138\n",
      "Reads with adapters:                        71 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        60,138 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,078,974 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,078,709 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 71 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t939.7\t0\t20\n",
      "4\t50\t234.9\t0\t50\n",
      "5\t1\t58.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327356_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327356_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327356_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327356_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        80,816 reads @   4.3 ¬µs/read;  13.97 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  80,816\n",
      "Reads with adapters:                        77 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        80,816 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,199,583 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,199,295 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 77 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1262.8\t0\t20\n",
      "4\t57\t315.7\t0\t57\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327360_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327360_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327360_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327360_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        68,936 reads @   4.9 ¬µs/read;  12.23 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,936\n",
      "Reads with adapters:                        79 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,936 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,395,340 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,395,049 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t1077.1\t0\t29\n",
      "4\t46\t269.3\t0\t46\n",
      "5\t4\t67.3\t0\t4\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327361_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327361_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327361_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327361_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        94,761 reads @   4.3 ¬µs/read;  13.97 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  94,761\n",
      "Reads with adapters:                        98 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        94,761 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,083,501 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,083,134 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 98 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1480.6\t0\t27\n",
      "4\t69\t370.2\t0\t69\n",
      "5\t2\t92.5\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327362_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327362_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327362_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327362_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        74,460 reads @   4.4 ¬µs/read;  13.77 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  74,460\n",
      "Reads with adapters:                        87 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        74,460 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,233,900 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,233,571 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 87 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t1163.4\t0\t19\n",
      "4\t68\t290.9\t0\t68\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327363_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327363_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327363_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327363_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        30,330 reads @   4.3 ¬µs/read;  14.11 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  30,330\n",
      "Reads with adapters:                        26 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        30,330 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,548,505 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,548,407 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 26 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t473.9\t0\t6\n",
      "4\t20\t118.5\t0\t20\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327364_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327364_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327364_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327364_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        76,923 reads @   4.3 ¬µs/read;  13.84 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  76,923\n",
      "Reads with adapters:                        81 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        76,923 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,614,455 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,614,154 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 81 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t25\t1201.9\t0\t25\n",
      "4\t55\t300.5\t0\t55\n",
      "6\t1\t18.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327367_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327367_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327367_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327367_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        79,786 reads @   4.2 ¬µs/read;  14.23 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  79,786\n",
      "Reads with adapters:                        73 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        79,786 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,045,299 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,045,024 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 73 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t1246.7\t0\t19\n",
      "4\t53\t311.7\t0\t53\n",
      "6\t1\t19.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327370_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327370_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327370_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327370_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        66,276 reads @   4.5 ¬µs/read;  13.33 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  66,276\n",
      "Reads with adapters:                        63 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        66,276 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,998,884 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,998,649 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 63 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t1035.6\t0\t19\n",
      "4\t42\t258.9\t0\t42\n",
      "5\t2\t64.7\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327379_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327379_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327379_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327379_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        65,516 reads @   4.3 ¬µs/read;  13.95 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  65,516\n",
      "Reads with adapters:                        75 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        65,516 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,891,178 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,890,900 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 75 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1023.7\t0\t27\n",
      "4\t45\t255.9\t0\t45\n",
      "5\t2\t64.0\t0\t2\n",
      "7\t1\t4.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327387_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327387_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327387_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327387_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        71,575 reads @   4.8 ¬µs/read;  12.57 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  71,575\n",
      "Reads with adapters:                        60 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        71,575 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,804,836 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,804,612 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 60 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t1118.4\t0\t16\n",
      "4\t44\t279.6\t0\t44\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327388_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327388_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327388_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327388_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        90,490 reads @   4.3 ¬µs/read;  14.06 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  90,490\n",
      "Reads with adapters:                       100 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        90,490 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,661,043 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,660,663 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 100 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1413.9\t0\t20\n",
      "4\t80\t353.5\t0\t80\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327394_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327394_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327394_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327394_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00       102,416 reads @   4.6 ¬µs/read;  13.10 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 102,416\n",
      "Reads with adapters:                        92 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):       102,416 (100.0%)\n",
      "\n",
      "Total basepairs processed:    15,462,993 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     15,462,643 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 92 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t1600.2\t0\t18\n",
      "4\t74\t400.1\t0\t74\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327395_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327395_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327395_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327395_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        52,488 reads @   5.0 ¬µs/read;  12.07 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  52,488\n",
      "Reads with adapters:                        59 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        52,488 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,921,956 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,921,735 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 59 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t820.1\t0\t15\n",
      "4\t44\t205.0\t0\t44\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327396_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327396_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327396_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327396_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        50,570 reads @   4.3 ¬µs/read;  13.85 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  50,570\n",
      "Reads with adapters:                        41 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        50,570 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,627,564 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,627,406 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 41 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t790.2\t0\t6\n",
      "4\t35\t197.5\t0\t35\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327401_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327401_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327401_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327401_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        58,412 reads @   4.4 ¬µs/read;  13.51 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  58,412\n",
      "Reads with adapters:                        81 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        58,412 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,767,598 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,767,293 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 81 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t912.7\t0\t19\n",
      "4\t62\t228.2\t0\t62\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327402_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327402_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327402_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327402_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        99,164 reads @   4.2 ¬µs/read;  14.26 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  99,164\n",
      "Reads with adapters:                        87 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        99,164 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,844,161 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,843,829 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 87 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t25\t1549.4\t0\t25\n",
      "4\t56\t387.4\t0\t56\n",
      "5\t3\t96.8\t0\t3\n",
      "6\t3\t24.2\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327404_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327404_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327404_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327404_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        84,245 reads @   4.2 ¬µs/read;  14.15 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  84,245\n",
      "Reads with adapters:                        97 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        84,245 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,718,727 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,718,354 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 97 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t1316.3\t0\t15\n",
      "4\t82\t329.1\t0\t82\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327405_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327405_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327405_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327405_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        30,801 reads @   4.4 ¬µs/read;  13.54 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  30,801\n",
      "Reads with adapters:                        36 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        30,801 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,649,995 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,649,864 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 36 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t481.3\t0\t13\n",
      "4\t23\t120.3\t0\t23\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327406_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327406_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327406_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327406_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        44,259 reads @   4.2 ¬µs/read;  14.30 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  44,259\n",
      "Reads with adapters:                        36 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        44,259 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,682,223 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,682,090 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 36 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t691.5\t0\t11\n",
      "4\t25\t172.9\t0\t25\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327407_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327407_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327407_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327407_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        68,172 reads @   4.5 ¬µs/read;  13.41 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,172\n",
      "Reads with adapters:                        60 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,172 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,243,846 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,243,620 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 60 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t1065.2\t0\t14\n",
      "4\t46\t266.3\t0\t46\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327408_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327408_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327408_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327408_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        68,859 reads @   4.4 ¬µs/read;  13.61 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,859\n",
      "Reads with adapters:                        84 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,859 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,338,270 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,337,939 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 84 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t1075.9\t0\t13\n",
      "4\t67\t269.0\t0\t67\n",
      "5\t2\t67.2\t0\t2\n",
      "7\t2\t4.2\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327409_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327409_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327409_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327409_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        74,939 reads @   4.2 ¬µs/read;  14.42 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  74,939\n",
      "Reads with adapters:                        62 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        74,939 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,314,399 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,314,168 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 62 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t1170.9\t0\t18\n",
      "4\t43\t292.7\t0\t43\n",
      "5\t1\t73.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327410_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327410_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327410_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327410_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,852 reads @   4.7 ¬µs/read;  12.70 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,852\n",
      "Reads with adapters:                        81 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,852 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,354,121 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,353,820 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 81 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t24\t1185.2\t0\t24\n",
      "4\t56\t296.3\t0\t56\n",
      "5\t1\t74.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327412_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327412_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327412_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327412_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        66,614 reads @   4.7 ¬µs/read;  12.75 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  66,614\n",
      "Reads with adapters:                        79 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        66,614 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,053,162 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,052,867 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 79 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t21\t1040.8\t0\t21\n",
      "4\t58\t260.2\t0\t58\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327414_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327414_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327414_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327414_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,293 reads @   4.3 ¬µs/read;  14.02 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,293\n",
      "Reads with adapters:                        73 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,293 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,368,317 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,368,053 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 73 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t1176.5\t0\t29\n",
      "4\t43\t294.1\t0\t43\n",
      "5\t1\t73.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327415_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327415_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327415_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327415_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,568 reads @   4.2 ¬µs/read;  14.16 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,568\n",
      "Reads with adapters:                        68 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,568 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,408,880 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,408,620 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 68 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t1180.8\t0\t17\n",
      "4\t47\t295.2\t0\t47\n",
      "5\t3\t73.8\t0\t3\n",
      "6\t1\t18.4\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327416_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327416_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327416_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327416_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        76,891 reads @   4.5 ¬µs/read;  13.39 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  76,891\n",
      "Reads with adapters:                        74 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        76,891 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,417,587 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,417,308 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 74 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1201.4\t0\t20\n",
      "4\t52\t300.4\t0\t52\n",
      "5\t1\t75.1\t0\t1\n",
      "6\t1\t18.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327418_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327418_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327418_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327418_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        35,629 reads @   4.4 ¬µs/read;  13.52 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  35,629\n",
      "Reads with adapters:                        43 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        35,629 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,377,033 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,376,872 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t556.7\t0\t11\n",
      "4\t32\t139.2\t0\t32\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327419_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327419_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327419_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327419_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        72,094 reads @   4.4 ¬µs/read;  13.63 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  72,094\n",
      "Reads with adapters:                        84 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        72,094 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,883,890 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,883,565 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 84 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t1126.5\t0\t13\n",
      "4\t69\t281.6\t0\t69\n",
      "5\t2\t70.4\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327421_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327421_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327421_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327421_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        74,580 reads @   4.3 ¬µs/read;  13.99 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  74,580\n",
      "Reads with adapters:                        86 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        74,580 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,213,025 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,212,697 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 86 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1165.3\t0\t20\n",
      "4\t63\t291.3\t0\t63\n",
      "5\t2\t72.8\t0\t2\n",
      "6\t1\t18.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327426_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327426_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327426_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327426_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        40,583 reads @   4.6 ¬µs/read;  13.04 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,583\n",
      "Reads with adapters:                        34 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        40,583 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,127,313 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,127,189 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 34 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12\t634.1\t0\t12\n",
      "4\t22\t158.5\t0\t22\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327427_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327427_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327427_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327427_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        58,261 reads @   4.4 ¬µs/read;  13.61 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  58,261\n",
      "Reads with adapters:                        45 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        58,261 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,796,856 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,796,694 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 45 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t910.3\t0\t19\n",
      "4\t25\t227.6\t0\t25\n",
      "5\t1\t56.9\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327428_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327428_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327428_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327428_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        43,041 reads @   4.2 ¬µs/read;  14.12 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  43,041\n",
      "Reads with adapters:                        27 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        43,041 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,497,981 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,497,879 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 27 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t672.5\t0\t6\n",
      "4\t21\t168.1\t0\t21\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327431_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327431_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327431_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327431_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        52,529 reads @   4.4 ¬µs/read;  13.57 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  52,529\n",
      "Reads with adapters:                        39 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        52,529 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,925,431 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,925,284 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 39 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t820.8\t0\t11\n",
      "4\t26\t205.2\t0\t26\n",
      "5\t2\t51.3\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327432_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327432_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327432_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327432_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        34,885 reads @   4.6 ¬µs/read;  12.98 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  34,885\n",
      "Reads with adapters:                        48 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        34,885 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,259,020 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,258,841 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 48 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t545.1\t0\t14\n",
      "4\t33\t136.3\t0\t33\n",
      "5\t1\t34.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327433_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327433_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327433_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327433_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        29,682 reads @   4.5 ¬µs/read;  13.24 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  29,682\n",
      "Reads with adapters:                        21 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        29,682 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,481,218 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,481,141 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 21 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t463.8\t0\t7\n",
      "4\t14\t115.9\t0\t14\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327434_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327434_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327434_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327434_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        48,495 reads @   4.4 ¬µs/read;  13.72 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  48,495\n",
      "Reads with adapters:                        48 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        48,495 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,277,684 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,277,499 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 48 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t7\t757.7\t0\t7\n",
      "4\t41\t189.4\t0\t41\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327435_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327435_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327435_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327435_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        61,016 reads @   4.3 ¬µs/read;  13.85 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  61,016\n",
      "Reads with adapters:                        74 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        61,016 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,212,391 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,212,110 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 74 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t953.4\t0\t16\n",
      "4\t57\t238.3\t0\t57\n",
      "5\t1\t59.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327439_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327439_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327439_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327439_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        73,356 reads @   4.3 ¬µs/read;  13.84 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  73,356\n",
      "Reads with adapters:                        72 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        73,356 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,038,384 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,038,125 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 72 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t1146.2\t0\t29\n",
      "4\t43\t286.5\t0\t43\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327442_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327442_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327442_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327442_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        59,270 reads @   4.4 ¬µs/read;  13.64 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  59,270\n",
      "Reads with adapters:                        55 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        59,270 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,927,616 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,927,414 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 55 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t926.1\t0\t20\n",
      "4\t33\t231.5\t0\t33\n",
      "5\t2\t57.9\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327443_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327443_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327443_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327443_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        54,309 reads @   4.8 ¬µs/read;  12.57 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  54,309\n",
      "Reads with adapters:                        44 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        54,309 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,147,080 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,146,916 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 44 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12\t848.6\t0\t12\n",
      "4\t32\t212.1\t0\t32\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327444_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327444_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327444_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327444_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        71,015 reads @   4.2 ¬µs/read;  14.14 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  71,015\n",
      "Reads with adapters:                        58 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        71,015 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,722,417 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,722,200 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 58 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t1109.6\t0\t16\n",
      "4\t41\t277.4\t0\t41\n",
      "5\t1\t69.4\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327445_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327445_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327445_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327445_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        28,933 reads @   4.3 ¬µs/read;  14.09 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  28,933\n",
      "Reads with adapters:                        32 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        28,933 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,367,686 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,367,563 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 32 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t452.1\t0\t6\n",
      "4\t25\t113.0\t0\t25\n",
      "5\t1\t28.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327447_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327447_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327447_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327447_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        74,103 reads @   4.2 ¬µs/read;  14.22 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  74,103\n",
      "Reads with adapters:                        62 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        74,103 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,187,397 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,187,165 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 62 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t1157.9\t0\t19\n",
      "4\t41\t289.5\t0\t41\n",
      "5\t1\t72.4\t0\t1\n",
      "6\t1\t18.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327450_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327450_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327450_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327450_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        72,829 reads @   4.5 ¬µs/read;  13.46 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  72,829\n",
      "Reads with adapters:                        64 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        72,829 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,988,922 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,988,687 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 64 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t23\t1138.0\t0\t23\n",
      "4\t39\t284.5\t0\t39\n",
      "5\t2\t71.1\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327452_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327452_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327452_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327452_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        79,803 reads @   4.2 ¬µs/read;  14.15 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  79,803\n",
      "Reads with adapters:                        70 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        79,803 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,047,979 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,047,712 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 70 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t1246.9\t0\t13\n",
      "4\t57\t311.7\t0\t57\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327456_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327456_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327456_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327456_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        62,282 reads @   4.6 ¬µs/read;  12.94 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  62,282\n",
      "Reads with adapters:                        30 (0.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        62,282 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,351,495 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,351,383 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 30 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t973.2\t0\t17\n",
      "4\t5\t243.3\t0\t5\n",
      "5\t7\t60.8\t0\t7\n",
      "6\t1\t15.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327464_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327464_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327464_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327464_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        65,517 reads @   4.6 ¬µs/read;  13.00 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  65,517\n",
      "Reads with adapters:                        37 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        65,517 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,763,806 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,763,675 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 37 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t1023.7\t0\t18\n",
      "4\t18\t255.9\t0\t18\n",
      "5\t1\t64.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327465_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327465_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327465_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327465_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        44,254 reads @   4.6 ¬µs/read;  13.03 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  44,254\n",
      "Reads with adapters:                        50 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        44,254 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,681,193 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,681,006 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 50 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t691.5\t0\t13\n",
      "4\t37\t172.9\t0\t37\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327470_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327470_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327470_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327470_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        97,454 reads @   4.6 ¬µs/read;  13.17 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  97,454\n",
      "Reads with adapters:                       125 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        97,454 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,690,983 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,690,507 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 125 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t31\t1522.7\t0\t31\n",
      "4\t91\t380.7\t0\t91\n",
      "6\t2\t23.8\t0\t2\n",
      "7\t1\t5.9\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327471_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327471_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327471_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327471_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        73,382 reads @   4.2 ¬µs/read;  14.29 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  73,382\n",
      "Reads with adapters:                        65 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        73,382 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,079,007 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,078,761 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 65 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t1146.6\t0\t14\n",
      "4\t51\t286.6\t0\t51\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327472_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327472_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327472_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327472_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,762 reads @   4.5 ¬µs/read;  13.46 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,762\n",
      "Reads with adapters:                        80 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,762 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,532,440 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,532,143 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 80 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t23\t1090.0\t0\t23\n",
      "4\t57\t272.5\t0\t57\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327475_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327475_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327475_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327475_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        37,342 reads @   4.3 ¬µs/read;  13.90 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,342\n",
      "Reads with adapters:                        40 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        37,342 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,637,476 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,637,331 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 40 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t583.5\t0\t16\n",
      "4\t23\t145.9\t0\t23\n",
      "5\t1\t36.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327479_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327479_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327479_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327479_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        36,821 reads @   4.6 ¬µs/read;  13.07 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  36,821\n",
      "Reads with adapters:                        45 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        36,821 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,559,187 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,559,016 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 45 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t575.3\t0\t11\n",
      "4\t32\t143.8\t0\t32\n",
      "5\t2\t36.0\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327481_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327481_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327481_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327481_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        82,674 reads @   4.3 ¬µs/read;  14.02 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  82,674\n",
      "Reads with adapters:                        86 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        82,674 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,481,105 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,480,784 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 86 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1291.8\t0\t27\n",
      "4\t56\t322.9\t0\t56\n",
      "5\t2\t80.7\t0\t2\n",
      "6\t1\t20.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327482_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327482_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327482_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327482_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        54,574 reads @   4.5 ¬µs/read;  13.29 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  54,574\n",
      "Reads with adapters:                        54 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        54,574 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,204,571 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,204,367 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 54 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t852.7\t0\t13\n",
      "4\t40\t213.2\t0\t40\n",
      "5\t1\t53.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327484_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327484_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327484_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327484_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        50,182 reads @   4.3 ¬µs/read;  14.03 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  50,182\n",
      "Reads with adapters:                        54 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        50,182 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,576,715 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,576,508 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 54 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t12\t784.1\t0\t12\n",
      "4\t40\t196.0\t0\t40\n",
      "5\t1\t49.0\t0\t1\n",
      "6\t1\t12.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327485_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327485_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327485_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327485_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        72,047 reads @   4.5 ¬µs/read;  13.44 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  72,047\n",
      "Reads with adapters:                        74 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        72,047 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,850,890 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,850,614 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 74 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t22\t1125.7\t0\t22\n",
      "4\t51\t281.4\t0\t51\n",
      "6\t1\t17.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327487_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327487_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327487_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327487_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        82,670 reads @   4.4 ¬µs/read;  13.63 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  82,670\n",
      "Reads with adapters:                       103 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        82,670 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,402,671 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,402,288 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 103 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t31\t1291.7\t0\t31\n",
      "4\t71\t322.9\t0\t71\n",
      "6\t1\t20.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327488_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327488_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327488_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327488_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        57,435 reads @   4.4 ¬µs/read;  13.73 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  57,435\n",
      "Reads with adapters:                        73 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        57,435 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,671,084 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,670,809 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 73 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t897.4\t0\t17\n",
      "4\t56\t224.4\t0\t56\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327489_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327489_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327489_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327489_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        57,039 reads @   4.5 ¬µs/read;  13.46 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  57,039\n",
      "Reads with adapters:                        57 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        57,039 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,563,128 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,562,917 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 57 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t891.2\t0\t19\n",
      "4\t37\t222.8\t0\t37\n",
      "6\t1\t13.9\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327492_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327492_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327492_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327492_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        89,728 reads @   4.2 ¬µs/read;  14.20 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  89,728\n",
      "Reads with adapters:                        71 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        89,728 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,474,554 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,474,283 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 71 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t15\t1402.0\t0\t15\n",
      "4\t54\t350.5\t0\t54\n",
      "5\t2\t87.6\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327493_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327493_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327493_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327493_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00       109,520 reads @   4.4 ¬µs/read;  13.70 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 109,520\n",
      "Reads with adapters:                       100 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):       109,520 (100.0%)\n",
      "\n",
      "Total basepairs processed:    16,449,180 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     16,448,801 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 100 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1711.2\t0\t27\n",
      "4\t69\t427.8\t0\t69\n",
      "5\t3\t107.0\t0\t3\n",
      "7\t1\t6.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327498_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327498_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327498_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327498_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        82,526 reads @   4.2 ¬µs/read;  14.30 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  82,526\n",
      "Reads with adapters:                        84 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        82,526 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,460,128 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,459,819 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 84 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1289.5\t0\t27\n",
      "4\t57\t322.4\t0\t57\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327502_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327502_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327502_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327502_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        79,707 reads @   4.2 ¬µs/read;  14.31 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  79,707\n",
      "Reads with adapters:                        88 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        79,707 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,019,804 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,019,481 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 88 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t1245.4\t0\t29\n",
      "4\t59\t311.4\t0\t59\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327503_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327503_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327503_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327503_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        54,585 reads @   4.3 ¬µs/read;  13.84 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  54,585\n",
      "Reads with adapters:                        56 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        54,585 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,002,132 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,001,921 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 56 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t852.9\t0\t18\n",
      "4\t34\t213.2\t0\t34\n",
      "5\t3\t53.3\t0\t3\n",
      "6\t1\t13.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327506_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327506_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327506_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327506_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        29,782 reads @   4.5 ¬µs/read;  13.21 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  29,782\n",
      "Reads with adapters:                        40 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        29,782 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,474,181 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,474,041 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 40 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t465.3\t0\t20\n",
      "4\t20\t116.3\t0\t20\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327507_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327507_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327507_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327507_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        49,626 reads @   4.4 ¬µs/read;  13.54 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  49,626\n",
      "Reads with adapters:                        62 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        49,626 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,426,451 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,426,211 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 62 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t775.4\t0\t11\n",
      "4\t50\t193.9\t0\t50\n",
      "7\t1\t3.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327509_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327509_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327509_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327509_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        40,794 reads @   4.3 ¬µs/read;  13.80 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,794\n",
      "Reads with adapters:                        42 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        40,794 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,121,466 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,121,311 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 42 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t637.4\t0\t14\n",
      "4\t27\t159.4\t0\t27\n",
      "5\t1\t39.8\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327511_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327511_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327511_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327511_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,205 reads @   4.2 ¬µs/read;  14.27 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,205\n",
      "Reads with adapters:                        47 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,205 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,447,715 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,447,540 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 47 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t14\t1081.3\t0\t14\n",
      "4\t32\t270.3\t0\t32\n",
      "5\t1\t67.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327513_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327513_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327513_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327513_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        43,878 reads @   4.7 ¬µs/read;  12.79 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  43,878\n",
      "Reads with adapters:                        71 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        43,878 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,623,630 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,623,362 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 71 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t685.6\t0\t20\n",
      "4\t47\t171.4\t0\t47\n",
      "5\t4\t42.8\t0\t4\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327515_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327515_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327515_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327515_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00       104,005 reads @   4.2 ¬µs/read;  14.26 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 104,005\n",
      "Reads with adapters:                       100 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):       104,005 (100.0%)\n",
      "\n",
      "Total basepairs processed:    15,428,419 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     15,428,046 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 100 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t28\t1625.1\t0\t28\n",
      "4\t71\t406.3\t0\t71\n",
      "5\t1\t101.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327516_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327516_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327516_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327516_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        69,958 reads @   4.5 ¬µs/read;  13.40 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  69,958\n",
      "Reads with adapters:                        59 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        69,958 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,562,229 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,562,010 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 59 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t19\t1093.1\t0\t19\n",
      "4\t38\t273.3\t0\t38\n",
      "5\t2\t68.3\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327517_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327517_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327517_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327517_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        74,689 reads @   4.4 ¬µs/read;  13.67 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  74,689\n",
      "Reads with adapters:                        81 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        74,689 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,267,786 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,267,478 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 81 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t1167.0\t0\t18\n",
      "4\t62\t291.8\t0\t62\n",
      "6\t1\t18.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327521_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327521_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327521_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327521_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        88,155 reads @   4.3 ¬µs/read;  13.80 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  88,155\n",
      "Reads with adapters:                       106 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        88,155 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,310,157 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,309,775 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 106 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t44\t1377.4\t0\t44\n",
      "4\t60\t344.4\t0\t60\n",
      "5\t2\t86.1\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327523_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327523_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327523_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327523_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        45,375 reads @   4.6 ¬µs/read;  13.13 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  45,375\n",
      "Reads with adapters:                        18 (0.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        45,375 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,850,421 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,850,353 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 18 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t4\t709.0\t0\t4\n",
      "4\t14\t177.2\t0\t14\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327529_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327529_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327529_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327529_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        25,686 reads @   4.5 ¬µs/read;  13.23 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  25,686\n",
      "Reads with adapters:                        27 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        25,686 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,857,903 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      3,857,800 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 27 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t401.3\t0\t5\n",
      "4\t22\t100.3\t0\t22\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327531_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327531_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327531_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327531_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        30,496 reads @   4.7 ¬µs/read;  12.63 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  30,496\n",
      "Reads with adapters:                        56 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        30,496 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,542,113 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,541,905 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 56 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t23\t476.5\t0\t23\n",
      "4\t29\t119.1\t0\t29\n",
      "5\t3\t29.8\t0\t3\n",
      "8\t1\t0.5\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327533_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327533_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327533_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327533_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        95,475 reads @   4.5 ¬µs/read;  13.44 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  95,475\n",
      "Reads with adapters:                       107 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        95,475 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,403,001 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,402,603 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 107 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t30\t1491.8\t0\t30\n",
      "4\t77\t372.9\t0\t77\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327534_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327534_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327534_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327534_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        44,177 reads @   4.4 ¬µs/read;  13.68 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  44,177\n",
      "Reads with adapters:                        69 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        44,177 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,668,597 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,668,338 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 69 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t690.3\t0\t17\n",
      "4\t52\t172.6\t0\t52\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327535_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327535_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327535_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327535_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        70,943 reads @   4.4 ¬µs/read;  13.75 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  70,943\n",
      "Reads with adapters:                        64 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        70,943 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,671,519 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,671,282 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 64 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t22\t1108.5\t0\t22\n",
      "4\t40\t277.1\t0\t40\n",
      "5\t1\t69.3\t0\t1\n",
      "6\t1\t17.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327539_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327539_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327539_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327539_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        89,026 reads @   4.5 ¬µs/read;  13.37 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  89,026\n",
      "Reads with adapters:                        98 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        89,026 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,337,070 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,336,705 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 98 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t30\t1391.0\t0\t30\n",
      "4\t66\t347.8\t0\t66\n",
      "5\t1\t86.9\t0\t1\n",
      "6\t1\t21.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327540_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327540_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327540_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327540_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        80,051 reads @   4.4 ¬µs/read;  13.65 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  80,051\n",
      "Reads with adapters:                        77 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        80,051 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,017,370 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,017,081 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 77 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1250.8\t0\t20\n",
      "4\t56\t312.7\t0\t56\n",
      "5\t1\t78.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327541_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327541_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327541_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327541_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        96,470 reads @   4.4 ¬µs/read;  13.65 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  96,470\n",
      "Reads with adapters:                       101 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        96,470 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,564,909 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,564,542 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 101 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t37\t1507.3\t0\t37\n",
      "4\t64\t376.8\t0\t64\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327544_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327544_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327544_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327544_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        51,563 reads @   4.5 ¬µs/read;  13.32 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  51,563\n",
      "Reads with adapters:                        51 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        51,563 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,785,302 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,785,106 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t805.7\t0\t10\n",
      "4\t39\t201.4\t0\t39\n",
      "5\t2\t50.4\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327546_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327546_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327546_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327546_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        76,765 reads @   4.5 ¬µs/read;  13.22 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  76,765\n",
      "Reads with adapters:                        52 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        76,765 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,422,480 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,422,287 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 52 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t17\t1199.5\t0\t17\n",
      "4\t33\t299.9\t0\t33\n",
      "5\t2\t75.0\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327550_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327550_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327550_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327550_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        37,365 reads @   4.7 ¬µs/read;  12.90 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  37,365\n",
      "Reads with adapters:                        43 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        37,365 (100.0%)\n",
      "\n",
      "Total basepairs processed:     5,640,694 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      5,640,532 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t13\t583.8\t0\t13\n",
      "4\t28\t146.0\t0\t28\n",
      "5\t1\t36.5\t0\t1\n",
      "6\t1\t9.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327554_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327554_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327554_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327554_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00       100,331 reads @   4.4 ¬µs/read;  13.72 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 100,331\n",
      "Reads with adapters:                       113 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):       100,331 (100.0%)\n",
      "\n",
      "Total basepairs processed:    15,147,852 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     15,147,426 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 113 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t37\t1567.7\t0\t37\n",
      "4\t68\t391.9\t0\t68\n",
      "5\t6\t98.0\t0\t6\n",
      "6\t1\t24.5\t0\t1\n",
      "7\t1\t6.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327555_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327555_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327555_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327555_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        27,971 reads @   4.4 ¬µs/read;  13.66 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  27,971\n",
      "Reads with adapters:                        36 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        27,971 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,222,917 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,222,776 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 36 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t437.0\t0\t5\n",
      "4\t29\t109.3\t0\t29\n",
      "5\t2\t27.3\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327557_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327557_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327557_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327557_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        20,785 reads @   4.5 ¬µs/read;  13.25 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  20,785\n",
      "Reads with adapters:                        23 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        20,785 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,136,642 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      3,136,558 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 23 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t8\t324.8\t0\t8\n",
      "4\t15\t81.2\t0\t15\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327561_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327561_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327561_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327561_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        62,580 reads @   4.3 ¬µs/read;  14.10 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  62,580\n",
      "Reads with adapters:                       112 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        62,580 (100.0%)\n",
      "\n",
      "Total basepairs processed:     9,447,177 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      9,446,748 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 112 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t977.8\t0\t20\n",
      "4\t91\t244.5\t0\t91\n",
      "5\t1\t61.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327562_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327562_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327562_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327562_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        56,059 reads @   4.5 ¬µs/read;  13.47 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  56,059\n",
      "Reads with adapters:                        66 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        56,059 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,463,071 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,462,815 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 66 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t11\t875.9\t0\t11\n",
      "4\t53\t219.0\t0\t53\n",
      "5\t1\t54.7\t0\t1\n",
      "6\t1\t13.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327566_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327566_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327566_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327566_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        21,386 reads @   4.6 ¬µs/read;  13.01 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  21,386\n",
      "Reads with adapters:                        31 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        21,386 (100.0%)\n",
      "\n",
      "Total basepairs processed:     3,228,280 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      3,228,164 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 31 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t9\t334.2\t0\t9\n",
      "4\t21\t83.5\t0\t21\n",
      "5\t1\t20.9\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327567_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327567_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327567_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327567_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        82,198 reads @   4.2 ¬µs/read;  14.24 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  82,198\n",
      "Reads with adapters:                        90 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        82,198 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,409,535 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,409,202 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 90 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t27\t1284.3\t0\t27\n",
      "4\t63\t321.1\t0\t63\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327568_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327568_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327568_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327568_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        50,174 reads @   4.5 ¬µs/read;  13.20 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  50,174\n",
      "Reads with adapters:                        25 (0.0%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        50,174 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,574,815 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,574,720 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 25 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t784.0\t0\t5\n",
      "4\t20\t196.0\t0\t20\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327569_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327569_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327569_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327569_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        87,957 reads @   4.3 ¬µs/read;  14.05 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  87,957\n",
      "Reads with adapters:                        60 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        87,957 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,279,018 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,278,793 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 60 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t1374.3\t0\t20\n",
      "4\t38\t343.6\t0\t38\n",
      "6\t1\t21.5\t0\t1\n",
      "7\t1\t5.4\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327573_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327573_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327573_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327573_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        49,475 reads @   4.4 ¬µs/read;  13.79 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  49,475\n",
      "Reads with adapters:                        51 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        49,475 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,390,382 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,390,195 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 51 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t773.0\t0\t18\n",
      "4\t32\t193.3\t0\t32\n",
      "5\t1\t48.3\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327575_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327575_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327575_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327575_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        40,015 reads @   4.8 ¬µs/read;  12.38 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,015\n",
      "Reads with adapters:                        40 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        40,015 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,004,789 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,004,633 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 40 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t5\t625.2\t0\t5\n",
      "4\t34\t156.3\t0\t34\n",
      "5\t1\t39.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327576_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327576_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327576_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327576_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        59,337 reads @   4.4 ¬µs/read;  13.74 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  59,337\n",
      "Reads with adapters:                        85 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        59,337 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,958,228 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,957,909 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 85 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t29\t927.1\t0\t29\n",
      "4\t51\t231.8\t0\t51\n",
      "5\t3\t57.9\t0\t3\n",
      "6\t1\t14.5\t0\t1\n",
      "7\t1\t3.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327581_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327581_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327581_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327581_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00       113,461 reads @   5.3 ¬µs/read;  11.24 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                 113,461\n",
      "Reads with adapters:                       119 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):       113,461 (100.0%)\n",
      "\n",
      "Total basepairs processed:    17,042,207 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     17,041,758 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 119 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t30\t1772.8\t0\t30\n",
      "4\t87\t443.2\t0\t87\n",
      "5\t1\t110.8\t0\t1\n",
      "6\t1\t27.7\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327584_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327584_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327584_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327584_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,846 reads @   4.4 ¬µs/read;  13.72 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,846\n",
      "Reads with adapters:                        78 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,846 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,378,732 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,378,439 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 78 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t21\t1185.1\t0\t21\n",
      "4\t55\t296.3\t0\t55\n",
      "5\t2\t74.1\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327586_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327586_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327586_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327586_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        75,404 reads @   4.2 ¬µs/read;  14.45 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  75,404\n",
      "Reads with adapters:                        80 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        75,404 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,276,994 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,276,691 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 80 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t1178.2\t0\t18\n",
      "4\t61\t294.5\t0\t61\n",
      "5\t1\t73.6\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327587_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327587_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327587_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327587_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        92,941 reads @   4.3 ¬µs/read;  13.89 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  92,941\n",
      "Reads with adapters:                        90 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        92,941 (100.0%)\n",
      "\n",
      "Total basepairs processed:    13,977,211 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     13,976,872 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 90 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t23\t1452.2\t0\t23\n",
      "4\t65\t363.1\t0\t65\n",
      "5\t2\t90.8\t0\t2\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327591_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327591_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327591_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327591_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        40,003 reads @   4.6 ¬µs/read;  12.91 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  40,003\n",
      "Reads with adapters:                        66 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        40,003 (100.0%)\n",
      "\n",
      "Total basepairs processed:     6,038,700 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      6,038,455 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 66 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t20\t625.0\t0\t20\n",
      "4\t45\t156.3\t0\t45\n",
      "5\t1\t39.1\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327592_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327592_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327592_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327592_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        80,107 reads @   4.3 ¬µs/read;  13.96 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  80,107\n",
      "Reads with adapters:                        70 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        80,107 (100.0%)\n",
      "\n",
      "Total basepairs processed:    12,029,643 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     12,029,384 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 70 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t22\t1251.7\t0\t22\n",
      "4\t47\t312.9\t0\t47\n",
      "5\t1\t78.2\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327596_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327596_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327596_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327596_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        26,675 reads @   4.6 ¬µs/read;  12.98 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  26,675\n",
      "Reads with adapters:                        43 (0.2%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        26,675 (100.0%)\n",
      "\n",
      "Total basepairs processed:     4,026,327 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      4,026,164 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 43 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t10\t416.8\t0\t10\n",
      "4\t32\t104.2\t0\t32\n",
      "5\t1\t26.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327599_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327599_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327599_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327599_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        68,336 reads @   4.3 ¬µs/read;  13.96 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  68,336\n",
      "Reads with adapters:                        71 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        68,336 (100.0%)\n",
      "\n",
      "Total basepairs processed:    10,317,121 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     10,316,850 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 71 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t16\t1067.8\t0\t16\n",
      "4\t52\t266.9\t0\t52\n",
      "5\t3\t66.7\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327604_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327604_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327604_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327604_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        48,728 reads @   4.2 ¬µs/read;  14.21 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  48,728\n",
      "Reads with adapters:                        38 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        48,728 (100.0%)\n",
      "\n",
      "Total basepairs processed:     7,357,221 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      7,357,075 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 38 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t6\t761.4\t0\t6\n",
      "4\t32\t190.3\t0\t32\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327605_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327605_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327605_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327605_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        76,774 reads @   4.2 ¬µs/read;  14.25 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  76,774\n",
      "Reads with adapters:                        80 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        76,774 (100.0%)\n",
      "\n",
      "Total basepairs processed:    11,590,752 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     11,590,456 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 80 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t25\t1199.6\t0\t25\n",
      "4\t54\t299.9\t0\t54\n",
      "5\t1\t75.0\t0\t1\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327615_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327615_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327615_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327615_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        93,805 reads @   4.3 ¬µs/read;  14.04 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  93,805\n",
      "Reads with adapters:                       121 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        93,805 (100.0%)\n",
      "\n",
      "Total basepairs processed:    14,128,892 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):     14,128,433 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 121 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t33\t1465.7\t0\t33\n",
      "4\t83\t366.4\t0\t83\n",
      "5\t2\t91.6\t0\t2\n",
      "6\t3\t22.9\t0\t3\n",
      "\n",
      "Command: cutadapt -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327620_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327620_01_L001_R1_001.fastq.gz\n",
      "\n",
      "This is cutadapt 5.1 with Python 3.10.14\n",
      "Command line parameters: -u 0 --error-rate 0.1 --times 1 --overlap 3 --minimum-length 1 -q 0,0 --quality-base 33 --cores 1 -o /tmp/qiime2/jovyan/processes/7501-1764698097.49@jovyan/tmp/q2-OutPath-7gdcsc5k/ERR5327620_01_L001_R1_001.fastq.gz --front CTTGGTCATTTAGAGGAAGTAA /tmp/qiime2/jovyan/data/3638611d-1767-413b-9390-70ee3d78e4ff/data/ERR5327620_01_L001_R1_001.fastq.gz\n",
      "Processing single-end reads on 1 core ...\n",
      "Done           00:00:00        53,136 reads @   4.3 ¬µs/read;  13.84 M reads/minute\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  53,136\n",
      "Reads with adapters:                        59 (0.1%)\n",
      "\n",
      "== Read fate breakdown ==\n",
      "Reads that were too short:                   0 (0.0%)\n",
      "Reads written (passing filters):        53,136 (100.0%)\n",
      "\n",
      "Total basepairs processed:     8,022,095 bp\n",
      "Quality-trimmed:                       0 bp (0.0%)\n",
      "Total written (filtered):      8,021,877 bp (100.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: CTTGGTCATTTAGAGGAAGTAA; Type: regular 5'; Length: 22; Trimmed: 59 times\n",
      "\n",
      "Minimum overlap: 3\n",
      "No. of allowed errors:\n",
      "1-9 bp: 0; 10-19 bp: 1; 20-22 bp: 2\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "3\t18\t830.2\t0\t18\n",
      "4\t41\t207.6\t0\t41\n",
      "\u001b[32mSaved SampleData[SequencesWithQuality] to: data/02/fungut_forward_reads_trimmed.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime cutadapt trim-single \\\n",
    "  --i-demultiplexed-sequences $data_dir/fungut_forward_reads.qza \\\n",
    "  --p-front CTTGGTCATTTAGAGGAAGTAA \\\n",
    "  --o-trimmed-sequences $data_dir/02/fungut_forward_reads_trimmed.qza \\\n",
    "  --verbose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "71d07265-297a-4597-95cf-bdb037054e64",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n! qiime demux summarize   --i-data $data_dir/02/fungut_forward_reads_trimmed.qza   --o-visualization $data_dir/02/demux_summary_posttrimming.qzv\\n'"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "! qiime demux summarize \\\n",
    "  --i-data $data_dir/02/fungut_forward_reads_trimmed.qza \\\n",
    "  --o-visualization $data_dir/02/demux_summary_posttrimming.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d57ddecf-4228-4283-9461-34586dee9b92",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/02/demux_summary_posttrimming.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c06f95f4-d775-495b-9eee-aa0a2cc08d16",
   "metadata": {},
   "source": [
    "<div style=\"background-color: lightblue; padding: 20px;\">\n",
    "Trimming our primers we can see that only XXXXXXXXX of feautres have been edited. Hence, we assume that the primers have already been trimmed as otherwise we would expect many more feautres to have been adjusted.\n",
    "</div>\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b853438-416a-4d66-a2ac-80eabfc1f842",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 03 Denoising üîä "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9a5402f-bb62-4b1b-b8be-5f848ed1edc7",
   "metadata": {},
   "source": [
    "Perfect! Now we're all set to start working with our demultiplexed and trimmed sequences. We start by denoising the sequences to recieve the holy feature table and the denoised sequences that will be the basis of all downstream processing.\n",
    "\n",
    "- Weli settings heimer brucht\n",
    "- Denoising benchmarking vor livia"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b01b127-58d6-4ea7-a239-272826ac1d25",
   "metadata": {},
   "source": [
    "## 03.01 Denoising benchmarking"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c94ad9c-ae7a-4bf4-9581-e336c09fb410",
   "metadata": {},
   "source": [
    "(LIVIA SCHRIBE) We tried out several denoising settings to see how they affect both sequence quality and the number of features we keep. To do so we ...."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6bf997a2-3a05-4249-8455-660ba3e89f38",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 03.02 Denoising our demultiplexed sequences using DADA2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8031c55-a0e2-4acf-a2e8-5f265cf3269d",
   "metadata": {
    "tags": []
   },
   "source": [
    "<div style=\"background-color: orange; padding: 20px;\">\n",
    "--p-max-ee 4 ‚Äî controls the maximum expected errors per read accepted by DADA2‚Äôs filtering. 4 is a relatively lenient filter; 1‚Äì2 is stricter and will remove more reads. Choose this based on the post-trim quality (look at the demux summary).\n",
    "<\\div>"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77e07106-7963-4bae-a685-231acbd3b821",
   "metadata": {},
   "source": [
    "XXXX why did we use the parameters we used?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "38eeb124-0686-4ed3-b7c2-7716b38fde87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#Denoising_1\\n! qiime dada2 denoise-single    --i-demultiplexed-seqs $data_dir/02/fungut_forward_reads_trimmed.qza    --p-trim-left 0    --p-trunc-len 0    --p-min-fold-parent-over-abundance 1    --p-max-ee 2     --o-representative-sequences $data_dir/03/denoising_1/dada2_rep_seqs_1.qza     --o-table $data_dir/03/denoising_1/dada2_table_1.qza     --o-denoising-stats $data_dir/03/denoising_1/dada2_stats_1.qza\\n'"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#Denoising_1\n",
    "! qiime dada2 denoise-single \\\n",
    "   --i-demultiplexed-seqs $data_dir/02/fungut_forward_reads_trimmed.qza \\\n",
    "   --p-trim-left 0 \\\n",
    "   --p-trunc-len 0 \\\n",
    "   --p-min-fold-parent-over-abundance 1 \\\n",
    "   --p-max-ee 2 \\\n",
    "    --o-representative-sequences $data_dir/03/denoising_1/dada2_rep_seqs_1.qza \\\n",
    "    --o-table $data_dir/03/denoising_1/dada2_table_1.qza \\\n",
    "    --o-denoising-stats $data_dir/03/denoising_1/dada2_stats_1.qza\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "8cf3d1e5-2571-47fe-8f03-bdd9ea990c87",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/03/denoising_4/dada2_table_4.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: data/03/denoising_4/dada2_rep_seqs_4.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[DADA2Stats] to: data/03/denoising_4/dada2_stats_4.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "#Denoising_4\n",
    "\n",
    "! qiime dada2 denoise-single \\\n",
    "   --i-demultiplexed-seqs $data_dir/02/fungut_forward_reads_trimmed.qza \\\n",
    "   --p-trim-left 0 \\\n",
    "   --p-trunc-len 0 \\\n",
    "   --p-min-fold-parent-over-abundance 4 \\\n",
    "   --p-max-ee 4 \\\n",
    "    --o-representative-sequences $data_dir/03/denoising_4/dada2_rep_seqs_4.qza \\\n",
    "    --o-table $data_dir/03/denoising_4/dada2_table_4.qza \\\n",
    "    --o-denoising-stats $data_dir/03/denoising_4/dada2_stats_4.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1def79a3-0d16-4586-aaa7-09b0cb8c4c60",
   "metadata": {
    "tags": []
   },
   "source": [
    "<div style=\"background-color: yellow; padding: 20px;\">\n",
    "After denoising, .....\n",
    "</div>\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82b02b51-f332-412a-9aa8-fd6173257d9e",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-10-15T09:32:46.029016Z",
     "iopub.status.busy": "2025-10-15T09:32:46.028062Z",
     "iopub.status.idle": "2025-10-15T09:32:46.032468Z",
     "shell.execute_reply": "2025-10-15T09:32:46.031780Z",
     "shell.execute_reply.started": "2025-10-15T09:32:46.028955Z"
    },
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 04 Taxonomy üçÑ‚Äçüü´ "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c74aae61-e25a-4f3a-9eff-42709bbb6127",
   "metadata": {},
   "source": [
    "XXXXXXX\n",
    "Great! Now that we have a finalised metadata file (for now...) and a denoised feature table, the last thing we need to make all the cool analyses later on is the taxonomic classification of our samples.\n",
    "\n",
    "To classify our samples, we used a pretrained classifier (INFOS √úBERE CLASSIFIER) and ran the script below (SCRIPT IRGENDWO IFE√®GE WOMER UF EULER BRUCHT HEI) on Euler to recieve a taxonomic classifiaction of our samples. In this notebook, we will first import the output file of the Euler-job and then we will first have a general look at our classification."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6b9304b9-b606-4dcf-a356-906ff63c0140",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 19:11:42--  https://polybox.ethz.ch/index.php/s/ybCKaH8Poyz372C/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 128973 (126K) [application/octet-stream]\n",
      "Saving to: ‚Äòdata/04/taxjob_euler.qza‚Äô\n",
      "\n",
      "data/04/taxjob_eule 100%[===================>] 125.95K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2025-12-02 19:11:43 (115 MB/s) - ‚Äòdata/04/taxjob_euler.qza‚Äô saved [128973/128973]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/ybCKaH8Poyz372C/download\" -O $data_dir/04/taxjob_euler.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82a79ab6-4703-4b87-9822-30f77b1c1201",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 04.01 Taxonomy overview "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9960ceca-002e-4064-b4c0-eb806c05f7bb",
   "metadata": {},
   "source": [
    "To have a first look at our taxonomic classification, we create the metadata table and the taxa barplots."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "8e70759e-9a19-4e62-ab99-530230ece0c2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/04/taxonomy_overview.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime metadata tabulate \\\n",
    "    --m-input-file $data_dir/04/taxjob_euler.qza \\\n",
    "    --o-visualization $data_dir/04/taxonomy_overview.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c1015248-d07e-418f-b469-0a30dcfcdbac",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/5ac1ecaa-fcfe-40c4-b9ec-4d4731ed7729')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 5ac1ecaa-fcfe-40c4-b9ec-4d4731ed7729>"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/04/taxonomy_overview.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7dacdece-4aed-49aa-b835-4f7fbf544283",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/04/taxa_bar_plots_overview.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime taxa barplot \\\n",
    "    --i-table $data_dir/03/denoising_4/dada2_table_4.qza \\\n",
    "    --i-taxonomy $data_dir/04/taxjob_euler.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed.tsv \\\n",
    "    --o-visualization $data_dir/04/taxa_bar_plots_overview.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "78324e4e-f1e8-4425-8b68-9aa04614d632",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/59eeece3-2c0b-4af9-9673-61c978f1567e')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 59eeece3-2c0b-4af9-9673-61c978f1567e>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/04/taxa_bar_plots_overview.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45c30fd1-e03d-4b53-92a7-a6225995fcef",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 04.02 Filtering our data to exclude any non-fungal features"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b67c6596-9c1b-4387-b43c-fd61c80bd54b",
   "metadata": {},
   "source": [
    "Looking at the taxonomic classification barplots, we see some non-fungal features and some samples with a high degree (>80%) of unclassifiable sequences. Let's have a look at some of these sequences using our metadata table and BLASTing the sequences. Here is an overview of randomly picked features:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa2f34a2-5541-43ee-a552-7b0993d122be",
   "metadata": {},
   "source": [
    "| sample ID | BLAST | Reason for filtering |\n",
    "|-------|-------|-------|\n",
    "| fed424ac80555019c8816c6efb6f2004   | Fungal endophyte isolate   |  No precise classification   |\n",
    "| 1ada978c8f3ff08af2668393869257ac    | Rhodotorula mucilaginosa  | Mostly from environmental sources | \n",
    "| 201dcc24ff7662a5eb05f31a0d0ca6c0    | no BLAST result found   |      |\n",
    "| 270002b21dcf309398fe1f945dc902e0    | uncultured fugus   |      |\n",
    "| 27e5e1e9ffac83de20b1ed2784cecee4    | Rhodotorula mucilaginosa   | Mostly from environmental sources |\n",
    "| 288c841878718e65bd3f27b674c5b16a    | Mucor racemosus   | Mostly from environmental sources  |\n",
    "| 31bd8178bf3013d48c62813a535b9e09 | Torulaspora delbrueckii |  Mostly from environmental sources |\n",
    "| 3e2d3eecabe4355bd15aeab85aebaa83 | Rhodotorula mucilaginosa |  Mostly from environmental sources |\n",
    "| 478e194c1ec118f79940439cc0c7fd36 | Cladosporium allicinum | plant leaves |\n",
    "| 4af6fc99c1acf99c1b82d1db3a538c9e | Uncultured fungus | |\n",
    "| 54e88722f1d30a88eaaf19efc686d249 | Uncultured basidiomycete |  |\n",
    "| 58de98eb759b336275f22b9a976e97e1 | Rhodotorula mucilaginosa | Mostly from environmental sources |\n",
    "| 5ad3902302653fe8bb0b938de1d18f68 | Uncultured fungus clone |  |\n",
    "| 68da1589cb5aa90483bf9ddded795ff5 | Uncultured fungus clone |  |\n",
    "| 73c2e92790e785ec39b58026ed4010ef | Candida glabrata |k__Fungi;p__Ascomycota;sp__Saccharomycotina;c__Saccharomycetes;o__Saccharomycetales;f__Saccharomycetaceae;g__Candida;s__glabrata\n",
    "  |\n",
    "| 818fb915612b44bbbd91b854c738a057 | Fungal endophyte | in plants |\n",
    "| 97d39fb074a634f6e55e91dd89e69c89 | Blastocystis | parasit |\n",
    "| a0dc22f3ab283a0afd30e730590fb85c | Uncultured fungus |  |\n",
    "| a2805006a5012b036fa721982c1324fb | Rhizopus oryzae strain |Mostly from environmental sources|\n",
    "| afbfe0c693e41ecdd14938a045832793 | Rhodotorula mucilaginosa |Mostly from environmental sources  |\n",
    "| b2cf5a969f4c7a2f29d49b4216a39244 | Kluyveromyces marxianus | Mostly from environmental sources |\n",
    "| beb8f130fd78d92d09b9aa1e37fec750 | Escherichia coli | Bacteria |\n",
    "| c6a89f484b775e3a8a0842ec9d7568e1 | Uncultured fungus |  |\n",
    "| c84004c0020de7d6ff5bde95677e8688 | Uncultured fungus |  |\n",
    "| c9da8911059b1b0a4a2980d229a06a69 | Candida albicans  | k__Fungi;p__Ascomycota;sp__Saccharomycotina;c__Saccharomycetes;o__Saccharomycetales;f__Debaryomycetaceae;g__Candida;s__albicans\n",
    " |\n",
    "| d8c7c2823a15d6d0e95a8003cf1cc15b | Rhodotorula mucilaginosa |Mostly from environmental sources  |\n",
    "| d94d771c3041404138de7df8e52c6cd1 | Aethes rutilana genome assembly | Metazoa |\n",
    "| ddf643cb96fee9d6ecd44d938b35bd61 | Uncultured fungus  |  |\n",
    "| eff5b3a232c5c37acde9e8fc390d2c12 | no BLAST result found|  |\n",
    "| f931ac9a7305cdd99a02c25a624e5bb8 | Pichia kudriavzevii  | Mostly from environmental sources |\n",
    "| fa340e00a973844b027665f54f7f6fb0 | Rhodotorula mucilaginosa  | Mostly from environmental sources |\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2fd73fbe-01fa-4c65-9a68-557001d9b2cf",
   "metadata": {},
   "source": [
    "As we can see from our randomly sampled unclassified features, almost all of the features stem from environmental sources or are from unclultured fungi. The features belonging to environmental samples and those belonging to plants and parasites are most likely contaminations that could distort downstream analyses. For this reason we decided to filter out any non-fungal features and only keep features that have have been classified to the level of class. We do this by using the qiime taxa filter-table/seqs commands:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "d8ba16b3-2e6c-4396-856f-d0000fde0f5d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/04/dada2_table_filtered.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureData[Sequence] to: data/04/dada2_rep_seqs_filtered.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime taxa filter-table \\\n",
    "    --i-table $data_dir/03/denoising_4/dada2_table_4.qza \\\n",
    "    --i-taxonomy $data_dir/04/taxjob_euler.qza \\\n",
    "    --p-exclude Protista,Eukaryota_kgd_Incertae_sedis,Ichthyosporia,Metazoa,Viridiplantae \\\n",
    "    --p-include p__ \\\n",
    "    --o-filtered-table $data_dir/04/dada2_table_filtered.qza\n",
    "\n",
    "! qiime taxa filter-seqs \\\n",
    "    --i-sequences $data_dir/03/denoising_4/dada2_rep_seqs_4.qza \\\n",
    "    --i-taxonomy $data_dir/04/taxjob_euler.qza \\\n",
    "    --p-exclude Protista,Eukaryota_kgd_Incertae_sedis,Ichthyosporia,Metazoa,Viridiplantae \\\n",
    "    --p-include p__ \\\n",
    "    --o-filtered-sequences $data_dir/04/dada2_rep_seqs_filtered.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "446c19d2-3093-4cbc-8199-fa78c0f24e7b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/03/denoising_4/dada2_table_4.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime feature-table summarize \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered.qza \\\n",
    "    --m-sample-metadata-file $data_dir/metadata_ed.tsv \\\n",
    "    --o-visualization $data_dir/03/denoising_4/dada2_table_4.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03b43eb4-91b1-4c64-a8ac-34d9bef23cf6",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 04.03 Removing samples high in unclassifiable sequences and high in non-fungi"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fbe9b8c-0297-4ace-8178-1446b8b2e835",
   "metadata": {},
   "source": [
    "In our taxonomy barplots, we also saw some samples that are very high (>80%) in unclassified sequences or non-fungal sequences. As they are made up to a large degree of either contaminants or unclassifiable fungi, we want to filter these out to make sure they don't impact our downstream processing. To filter those out, we adjust our metadata and remove the IDs of the unwanted sequences. Then, we create a new feature table summary to see how many samples we lose in the process of filtering."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b246f420-2d83-43c3-b47f-ff9ee3079590",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_ed = pd.read_csv(f\"{data_dir}/metadata_ed.tsv\", sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "574a97a3-2969-47e1-8ad0-dabd5648bbb9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "rem = [\"ERR5327575\", \"ERR5327509\", \"ERR5327351\", \"ERR5327544\", \"ERR5327300\", \"ERR5327338\", \"ERR5327529\", \"ERR5327533\", \"ERR5327364\", \"ERR5327535\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "aad50570-5819-4494-92c4-33fb81755948",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "metadata_ed_rem = metadata_ed[~metadata_ed[\"ID\"].isin(rem)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "23f1ba4f-d321-4b61-86a5-9cc33b227aab",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_ed_rem.to_csv(f\"{data_dir}/metadata_ed_rem.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "4dc2eac8-76b8-47ec-bfba-5015b48328ad",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/04/dada2_table_filtered_samples_removed.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime feature-table filter-samples \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-filtered-table $data_dir/04/dada2_table_filtered_samples_removed.qza "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "c857a320-e13c-4286-b732-fef998b0de7a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/04/dada2_table_metadata.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime feature-table summarize \\\n",
    "  --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "  --o-visualization $data_dir/04/dada2_table_metadata.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "4ba92e34-8985-4482-908c-2bee93db4938",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/be1f4e8f-8ae0-486c-9b9c-f411fa51ab96')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: be1f4e8f-8ae0-486c-9b9c-f411fa51ab96>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/04/dada2_table_metadata.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "989a7065-61ef-41ea-aa0d-eb982cae275c",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 04.04 Taxonomy after filtering"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c6f645fc-785d-4875-be7e-1045b8953d0d",
   "metadata": {},
   "source": [
    "In the feautre table summary from the last visualisation, we see that we have now have 140 samples, 10 less than what we started with. Cool, now let's see what has changed in the taxonomic classification after filtering out the unwanted sequences:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "9d5cd66a-cc0f-41af-add0-8390bb49db78",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/04/taxa_barplots_filtered.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime taxa barplot \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "    --i-taxonomy $data_dir/04/taxjob_euler.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/04/taxa_barplots_filtered.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6ec0a6e8-b9b9-4d86-b465-dd74cfe4cb83",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/cde5f7e1-ff96-4882-8d4b-d1a460a5e05f')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: cde5f7e1-ff96-4882-8d4b-d1a460a5e05f>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/04/taxa_barplots_filtered.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "47aa1133-9307-4ec7-ac69-ac168401d0ca",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 05 The path to an appropriate sequencing depth üìê"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "256d0f1a-f727-47cf-9171-adf0733c5fe0",
   "metadata": {
    "tags": []
   },
   "source": [
    "To ensure we comapre our samples fairly (or what the pros call \"statistically correct\"), we need to standerdize sample sizes by randomly selecting a certain number of sequences from each sample. This certain number is the sequencing depth (SD) and in the following steps, we will dtermine the SD based on the output of alpha rarefaction performed at different max depths."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c870168b-0cfc-4db7-b7f4-2f2e93219163",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 05.01 Finding an appropriate sequencing depth"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f825ec61-96cc-4d65-8f9a-36ae26880868",
   "metadata": {},
   "source": [
    "First we look at the output we get when we do alpha rarefaction at two smartly selected maximal depths (30'000 and 50'000) based on previous trials. For both depths, we see that both the shannon value plateaus very quickly (already at 1000 features for all metadata columns) and the observed features also pleateau early for most metadata columns. Comparing with how many samples we loose (we don't loose any until around 20'000), we decided on a sequencing depth of 20'000 as all curves have plateaued and we have not lost any samples yet."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "47df2eff-2afa-497f-9f03-297713f8f14b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#30000\\n#! qiime diversity alpha-rarefaction     --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\n    --p-max-depth 30000     --m-metadata-file $data_dir/metadata_ed_rem.tsv     --o-visualization $data_dir/05/fungut_alpha_rarefaction_30000.qzv\\n'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#30000\n",
    "#! qiime diversity alpha-rarefaction \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \n",
    "    --p-max-depth 30000 \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/05/fungut_alpha_rarefaction_30000.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "7c6d5a32-9e41-4da7-b186-89a43e77d99a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#50000\\n#! qiime diversity alpha-rarefaction     --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\n    --p-max-depth 50000     --m-metadata-file $data_dir/metadata_ed_rem.tsv     --o-visualization $data_dir/05/fungut_alpha_rarefaction_50000.qzv\\n'"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#50000\n",
    "#! qiime diversity alpha-rarefaction \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \n",
    "    --p-max-depth 50000 \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/05/fungut_alpha_rarefaction_50000.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "5dc6bced-0593-4963-855c-a317fb5d5284",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/05/fungut_alpha_rarefaction_30000.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e91966ee-bf38-4d41-a7cd-94c8e574c078",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/05/fungut_alpha_rarefaction_50000.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "06ec9c1b-22f6-460a-96d0-da62f75b0218",
   "metadata": {
    "tags": []
   },
   "source": [
    "In the following steps, we check how many samples we would lose fore three possible candidates of sequencing depths by filtering our filtered feature table (so much filtering going on eh?) and summarizing the output to see how many samples we retain."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6ead44c-de9d-4bcf-a685-9c10d6d0d1a3",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Sequencing depths to test (how many samples do we loose?):\n",
    "    1. 17500\n",
    "    2. 20000\n",
    "    3. 22500\n",
    "    \n",
    "Results:\n",
    "17500: no samples lost (150)\n",
    "20000: 1 sample lost (149)\n",
    "22500: 3 samples lost (147)\n",
    "\n",
    "-> Lets go with 20000. This way we have a good plateau for all curves and lose only 1 sample. \n",
    "-> We don't have any extremly low-depth samples that might need filtering (yay!)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "9471d0a5-3539-4a4b-9134-aad83d7db00a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#Test = 17500\\n!qiime feature-table filter-samples   --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza   --p-min-frequency 17500   --o-filtered-table $data_dir/05/table_minfreq_17500.qza\\n\\n!qiime feature-table summarize   --i-table $data_dir/05/table_minfreq_17500.qza   --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv   --o-visualization $data_dir/05/table_minfreq_17500_summary.qzv\\n'"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#Test = 17500\n",
    "!qiime feature-table filter-samples \\\n",
    "  --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "  --p-min-frequency 17500 \\\n",
    "  --o-filtered-table $data_dir/05/table_minfreq_17500.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table $data_dir/05/table_minfreq_17500.qza \\\n",
    "  --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/05/table_minfreq_17500_summary.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "f6995057-39a5-460c-826a-ff14dc922f0f",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#Depth 20000\\n!qiime feature-table filter-samples   --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza   --p-min-frequency 20000   --o-filtered-table $data_dir/05/table_minfreq_20000.qza\\n\\n!qiime feature-table summarize   --i-table $data_dir/06/table_minfreq_20000.qza   --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv   --o-visualization $data_dir/06/table_minfreq_20000_summary.qzv\\n'"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#Depth 20000\n",
    "!qiime feature-table filter-samples \\\n",
    "  --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "  --p-min-frequency 20000 \\\n",
    "  --o-filtered-table $data_dir/05/table_minfreq_20000.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table $data_dir/06/table_minfreq_20000.qza \\\n",
    "  --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/table_minfreq_20000_summary.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "c4173e5f-cbf8-4a3b-9ea2-f2f5312ac13a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#Depth 22500\\n!qiime feature-table filter-samples   --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza   --p-min-frequency 22500   --o-filtered-table $data_dir/05/table_minfreq_22500.qza\\n\\n!qiime feature-table summarize   --i-table $data_dir/05/table_minfreq_22500.qza   --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv   --o-visualization $data_dir/05/table_minfreq_22500_summary.qzv\\n'"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#Depth 22500\n",
    "!qiime feature-table filter-samples \\\n",
    "  --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "  --p-min-frequency 22500 \\\n",
    "  --o-filtered-table $data_dir/05/table_minfreq_22500.qza\n",
    "\n",
    "!qiime feature-table summarize \\\n",
    "  --i-table $data_dir/05/table_minfreq_22500.qza \\\n",
    "  --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/05/table_minfreq_22500_summary.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "b802f208-5973-402d-9d07-2ce3892d79a9",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/05/table_minfreq_17500_summary.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "664006d8-e3ca-4a93-8427-65f2a471c802",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/05/table_minfreq_20000_summary.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "b8ae76c7-04ec-459b-b590-e3b7eb549a63",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/06/table_minfreq_22500_summary.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fffe3a74-a66e-4276-aeec-970a6ae06a53",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 05.02 Final check before Bootstrapping"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "813a0de1-731a-4c1c-90dc-b9135cafc8b1",
   "metadata": {},
   "source": [
    "As a last quick check before bootstrapping and kmerizing our features, we perform alpha-rarefaction at a max depth of 23000 50 times to see if the randomness factor of alpha rarefaction might have lead to an unwanted bias. Luckily for us, we see no such effect in the output of this last check so we decided to continue with a SD of 20000."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "de96fc24-c1b7-4af5-a257-01164504f575",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n!qiime diversity alpha-rarefaction   --i-table $data_dir/04/dada2_table_filtered.qza   --p-max-depth 23000   --p-iterations 50   --m-metadata-file $data_dir/metadata_ed_rem.tsv   --o-visualization $data_dir/05/alpha_rarefaction_sanitycheck.qzv\\n'"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "!qiime diversity alpha-rarefaction \\\n",
    "  --i-table $data_dir/04/dada2_table_filtered.qza \\\n",
    "  --p-max-depth 23000 \\\n",
    "  --p-iterations 50 \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/05/alpha_rarefaction_sanitycheck.qzv\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "086108ce-959e-462f-8728-3829fa9c143c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#Visualization.load(f\"{data_dir}/05/alpha_rarefaction_sanitycheck.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93645bf1-2b5c-407f-a725-20fff992a750",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 06 Diversity analysis üìä"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae48e126-ae77-4f28-af6d-34f1c91c2993",
   "metadata": {
    "tags": []
   },
   "source": [
    "Aaaalright, now that we have our sequencing depth, we can continue analyzing our data and perform diversity analysis. To do so, we investigate both alpha diversity (=within-sample diversity) and beta diversity (=between sample diversity). \n",
    "\n",
    "For the actual computation, we rely on k-mer‚Äìbased diversity metrics. Recent puplications on ITS workflows show that k-mer methods outperform feature-table approaches by avoiding biases introduced during denoising and taxonomy assignment. In practice, this means we compute all diversity metrics directly from sequence-derived k-mers using qiime boots kmer-diversity. We perform this command at our previously selected sequencing depth and choose a kmer length of 16 after discussing the issue with our lovely TA. As we are running the bootstrapping at 1000 iterations, we must rely on the computational power of euler once again to run the job. Below, we will show the script we provided Euler and then we import the output directory using wget once more. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "03ec9085-fa46-4dd7-af76-eb81b93fb83d",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 06.01.01 Bootstrapping and kmerizing our ASVs on Euler "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7a4ddfc1-e808-4ef0-a76b-bacfdfa7f2b1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n\\n#!/bin/bash\\n#SBATCH --job-name=taxonomy_job\\n#SBATCH --ntasks=1\\n#SBATCH --cpus-per-task=1\\n#SBATCH --mem-per-cpu=128G\\n#SBATCH --time=24:00:00\\n#SBATCH --output=taxonomy_output_%j.txt\\n#SBATCH --mail-type=END\\n\\n\\nmodule load stack/2024-06 gcc/12.2.0 python/3.10\\n\\nsource ~/miniconda3/etc/profile.d/conda.sh\\nconda activate /cluster/scratch/lglauser/conda_envs/q2-boots-amplicon-2025.7\\nqiime dev refresh-cache\\n\\nqiime boots kmer-diversity   --i-table /cluster/home/lglauser/boot/dada2_table_filtered_samples_removed.qza   --i-sequences /cluster/home/lglauser/boot/dada2_rep_seqs_filtered.qza   --m-metadata-file /cluster/home/lglauser/boot/metadata_ed_rem.tsv   --p-sampling-depth 20000   --p-n 1000   --p-replacement   --p-kmer-size 16   --p-alpha-average-method median   --p-beta-average-method non-metric-median   --p-color-by \"country_sample\"   --output-dir /cluster/scratch/lglauser/boots_kmer_diversity_1000\\n  \\n'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#We ran the code below on euler to get our bootrstrapped diversity metrics.\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "#!/bin/bash\n",
    "#SBATCH --job-name=taxonomy_job\n",
    "#SBATCH --ntasks=1\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem-per-cpu=128G\n",
    "#SBATCH --time=24:00:00\n",
    "#SBATCH --output=taxonomy_output_%j.txt\n",
    "#SBATCH --mail-type=END\n",
    "\n",
    "\n",
    "module load stack/2024-06 gcc/12.2.0 python/3.10\n",
    "\n",
    "source ~/miniconda3/etc/profile.d/conda.sh\n",
    "conda activate /cluster/scratch/lglauser/conda_envs/q2-boots-amplicon-2025.7\n",
    "qiime dev refresh-cache\n",
    "\n",
    "qiime boots kmer-diversity \\\n",
    "  --i-table /cluster/home/lglauser/boot/dada2_table_filtered_samples_removed.qza \\\n",
    "  --i-sequences /cluster/home/lglauser/boot/dada2_rep_seqs_filtered.qza \\\n",
    "  --m-metadata-file /cluster/home/lglauser/boot/metadata_ed_rem.tsv \\\n",
    "  --p-sampling-depth 20000 \\\n",
    "  --p-n 1000 \\\n",
    "  --p-replacement \\\n",
    "  --p-kmer-size 16 \\\n",
    "  --p-alpha-average-method median \\\n",
    "  --p-beta-average-method non-metric-median \\\n",
    "  --p-color-by \"country_sample\" \\\n",
    "  --output-dir /cluster/scratch/lglauser/boots_kmer_diversity_1000\n",
    "  \n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8dcd23e-83c0-4fa8-a1fa-05faa97dcaf6",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 06.01.02 Importing our Euler results into the notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b4c05ed-0ac5-4eee-9294-88bee5328484",
   "metadata": {},
   "source": [
    "Now that we have run the job on euler, lets download the relevant outputs from the polybox as zip files and unzip them here to create our diversity analysis directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "77649121-7f62-47ad-bb75-4c658e2b1df4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 19:16:02--  https://polybox.ethz.ch/index.php/s/dZCGqG66fdafKGB/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 239231768 (228M) [application/zip]\n",
      "Saving to: ‚Äòdata/06/boots-kmer-diversity/alpha_diversities.zip‚Äô\n",
      "\n",
      "data/06/boots-kmer- 100%[===================>] 228.15M   142MB/s    in 1.6s    \n",
      "\n",
      "2025-12-02 19:16:04 (142 MB/s) - ‚Äòdata/06/boots-kmer-diversity/alpha_diversities.zip‚Äô saved [239231768/239231768]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/dZCGqG66fdafKGB/download\" -O $data_dir/06/boots-kmer-diversity/alpha_diversities.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "d0ebee01-2128-4fba-a063-5a89f2b23669",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  data/06/boots-kmer-diversity/alpha_diversities.zip\n",
      "   creating: data/06/boots-kmer-diversity/alpha_diversities/\n",
      " extracting: data/06/boots-kmer-diversity/alpha_diversities/observed_features.qza  \n",
      " extracting: data/06/boots-kmer-diversity/alpha_diversities/pielou_e.qza  \n",
      " extracting: data/06/boots-kmer-diversity/alpha_diversities/shannon.qza  \n"
     ]
    }
   ],
   "source": [
    "! unzip $data_dir/06/boots-kmer-diversity/alpha_diversities.zip -d $data_dir/06/boots-kmer-diversity/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "a3b23aac-5d23-414f-89db-9efa76fd3e24",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 19:16:06--  https://polybox.ethz.ch/index.php/s/Nppkb3p2GdgxjzN/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 159582191 (152M) [application/zip]\n",
      "Saving to: ‚Äòdata/06/boots-kmer-diversity/distance_matrices.zip‚Äô\n",
      "\n",
      "data/06/boots-kmer- 100%[===================>] 152.19M   154MB/s    in 1.0s    \n",
      "\n",
      "2025-12-02 19:16:07 (154 MB/s) - ‚Äòdata/06/boots-kmer-diversity/distance_matrices.zip‚Äô saved [159582191/159582191]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/Nppkb3p2GdgxjzN/download\" -O $data_dir/06/boots-kmer-diversity/distance_matrices.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "02d1d536-5a3a-4dd7-8886-740e446e507a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  data/06/boots-kmer-diversity/distance_matrices.zip\n",
      "   creating: data/06/boots-kmer-diversity/distance_matrices/\n",
      " extracting: data/06/boots-kmer-diversity/distance_matrices/braycurtis.qza  \n",
      " extracting: data/06/boots-kmer-diversity/distance_matrices/jaccard.qza  \n"
     ]
    }
   ],
   "source": [
    "! unzip $data_dir/06/boots-kmer-diversity/distance_matrices.zip -d $data_dir/06/boots-kmer-diversity/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "740cb682-09cd-4820-abe0-bf083d4834b8",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2025-12-02 19:16:09--  https://polybox.ethz.ch/index.php/s/arJ6ZLrxYYLJXwR/download\n",
      "Resolving polybox.ethz.ch (polybox.ethz.ch)... 129.132.71.243\n",
      "Connecting to polybox.ethz.ch (polybox.ethz.ch)|129.132.71.243|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 159578691 (152M) [application/zip]\n",
      "Saving to: ‚Äòdata/06/boots-kmer-diversity/pcoas.zip‚Äô\n",
      "\n",
      "data/06/boots-kmer- 100%[===================>] 152.19M   136MB/s    in 1.1s    \n",
      "\n",
      "2025-12-02 19:16:11 (136 MB/s) - ‚Äòdata/06/boots-kmer-diversity/pcoas.zip‚Äô saved [159578691/159578691]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!wget \"https://polybox.ethz.ch/index.php/s/arJ6ZLrxYYLJXwR/download\" -O $data_dir/06/boots-kmer-diversity/pcoas.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "c81f67dc-2bdf-476d-80a3-0ba68d21760f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  data/06/boots-kmer-diversity/pcoas.zip\n",
      "   creating: data/06/boots-kmer-diversity/pcoas/\n",
      " extracting: data/06/boots-kmer-diversity/pcoas/.order  \n",
      " extracting: data/06/boots-kmer-diversity/pcoas/braycurtis.qza  \n",
      " extracting: data/06/boots-kmer-diversity/pcoas/jaccard.qza  \n"
     ]
    }
   ],
   "source": [
    "! unzip $data_dir/06/boots-kmer-diversity/pcoas.zip -d $data_dir/06/boots-kmer-diversity/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac15f2b2-ca7c-4032-9630-7d776abd29df",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 06.02 Alpha diversity"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db680534-bd4f-4b4b-99ce-0e72f6997fdb",
   "metadata": {
    "tags": []
   },
   "source": [
    "### 06.02.01 Making and analyzing the plots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e58b19cb-ed75-411e-a60a-03e7fcf3af4e",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Categorical Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "3aeb8846-ff12-4fbf-9f6a-b65f31793a71",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_shannon_group_significance.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_observed_features_group_significance.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_pielou_e_group_significance.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/shannon.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_shannon_group_significance.qzv\n",
    "\n",
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/observed_features.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_observed_features_group_significance.qzv\n",
    "\n",
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/pielou_e.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_pielou_e_group_significance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0d501e94-fe4b-4a68-adcc-d95cd75a7850",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/39109161-7e74-4c1b-8b40-a87b7130f6c9')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 39109161-7e74-4c1b-8b40-a87b7130f6c9>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_shannon_group_significance.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "ed89bb6b-b420-4429-9128-c07cf00d7898",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/f70a43b2-ab0e-41c7-ac06-b1fd57066827')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f70a43b2-ab0e-41c7-ac06-b1fd57066827>"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_observed_features_group_significance.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a57e550b-ffd8-4072-93dd-09c1769d7292",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/324b5c34-681f-425a-810a-de4d2e9132a5')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 324b5c34-681f-425a-810a-de4d2e9132a5>"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_pielou_e_group_significance.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "91ea235a-ae9a-4f67-b9a9-16c5f696d5cb",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Numeric Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "35870d3a-d0cb-4f75-88ab-01a1116de9dc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_shannon_corrleation.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_observed_features_correlation.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/alpha_diversity/alpha_pielou_e_correlation.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-correlation \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/shannon.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_shannon_corrleation.qzv\n",
    "\n",
    "!qiime diversity alpha-correlation \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/observed_features.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_observed_features_correlation.qzv\n",
    "\n",
    "!qiime diversity alpha-correlation \\\n",
    "  --i-alpha-diversity $data_dir/06/boots-kmer-diversity/alpha_diversities/pielou_e.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/alpha_diversity/alpha_pielou_e_correlation.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "ed80735b-3e03-499e-a1dc-aba3d9dde2fa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/6bd4750c-190f-46f1-a7af-037dccf525be')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 6bd4750c-190f-46f1-a7af-037dccf525be>"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_shannon_corrleation.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "655b3369-5e59-4345-b60d-c249d5771141",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/83db5570-166f-4496-b59f-35640183b639')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 83db5570-166f-4496-b59f-35640183b639>"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_observed_features_correlation.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "0e3dbf65-bcaf-4961-9c1d-a29e0e207074",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/f4bcf00f-5e80-4715-a64f-426a30d4b199')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f4bcf00f-5e80-4715-a64f-426a30d4b199>"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/alpha_diversity/alpha_pielou_e_correlation.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd06be55-15c1-484f-8f43-cc5a96f3f84a",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### 06.02.2 Alpha diversity findings"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ccd54203-da5e-413b-a686-d2c769415be0",
   "metadata": {
    "tags": []
   },
   "source": [
    "<div style=\"background-color: yellow; padding: 20px;\">\n",
    "Analysing our alpha diversity plots for all three metrics of shannon, observed features and pielou we found the following things:\n",
    "</div>\n",
    "\n",
    "**Numeric variables:**\n",
    "- Age_years_sample is relevant only for richnenss (pValue=0.0174 in observed_feature alpha correlation)\n",
    "- All other numeric variables showed no significance for all three metrics\n",
    "\n",
    "**Categorical Variables:**\n",
    "- Statistical differnce found for **diet type**: both shannon and observed features show relevant p-Values (Shannon: 0.0273, observed features: 0.0274) for comaprisons to all other diet types (Vegan sample size n=5)\n",
    "- Statistical differnce found for **IDB status**: Significant p-Value (0.0119) for shannon metric when comparing participants with diagnosed IDB (n=10) to participants without (n=121).\n",
    "\n",
    "- No statistically relevant differences for the following variables:\n",
    "    - Country\n",
    "    - State\n",
    "    - Sex\n",
    "    - Celiac disease\n",
    "    - BMI\n",
    "    - Rural/Urban/peri"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eecd4fd9-6410-482e-9c62-f5e45b3e332b",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 06.03 Beta diversity"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7fb4573-144b-4aa6-9e5d-158032c418ea",
   "metadata": {},
   "source": [
    "XXX To investigate between-sample diversity (=beta diversity) we take the beta diversity outputs from the q2 boots kmer diversity directory and convert them to emperor plots. In the plots we can then search for any clustering of our samples (spoiler alert: we don't find any clear clustering)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42531df2-a013-4644-835e-3711f9394a82",
   "metadata": {
    "tags": []
   },
   "source": [
    "### 06.03.01 Making the plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "1dfa4fbb-74d5-4e5a-8bc8-9495ac416dfd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/beta_diversity_plots/kmer_bray_emperor.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/beta_diversity_plots/kmer_jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Bray‚ÄìCurtis\n",
    "!qiime emperor plot \\\n",
    "  --i-pcoa $data_dir/06/boots-kmer-diversity/pcoas/braycurtis.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/beta_diversity_plots/kmer_bray_emperor.qzv\n",
    "\n",
    "# Jaccard\n",
    "!qiime emperor plot \\\n",
    "  --i-pcoa $data_dir/06/boots-kmer-diversity/pcoas/jaccard.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/06/beta_diversity_plots/kmer_jaccard_emperor.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c198e754-7155-4f3c-9cd5-9f0014f8eac2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/4c3192c0-db35-48d2-870b-6658a2b740f3')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4c3192c0-db35-48d2-870b-6658a2b740f3>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/beta_diversity_plots/kmer_bray_emperor.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "13e1c107-22a6-4fcd-9006-86deae45d927",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/d7fc2163-1219-4e8c-ae2d-45532aa7c68e')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: d7fc2163-1219-4e8c-ae2d-45532aa7c68e>"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/beta_diversity_plots/kmer_jaccard_emperor.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2a967db-0fab-4d94-942d-e46bb4203883",
   "metadata": {
    "tags": []
   },
   "source": [
    "### 06.03.02 PERMANOVA"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "92366635-63af-4a85-9632-736c9b2009ea",
   "metadata": {},
   "source": [
    "XXXX Having found no clear clustering in our beta diversity plots, let us now check if we find statistically relevant differences between our samples using PERMANOVA. \n",
    "\n",
    "Variables of interest:\n",
    "- Diet type\n",
    "- IDB Status\n",
    "- BMI"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45ce433d-84bc-43d2-b4ad-9f5f815ed1ca",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Diet type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "5e322f96-ece9-4297-b669-b344f4fb4b26",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/permanova/permanova_bray_diet_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/permanova/permanova_jaccard_diet_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Pairwise Bray-Curtis\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/06/boots-kmer-diversity/distance_matrices/braycurtis.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column diet_type_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/06/permanova/permanova_bray_diet_pairwise.qzv\n",
    "\n",
    "# Pairwise Jaccard\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/06/boots-kmer-diversity/distance_matrices/jaccard.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column diet_type_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/06/permanova/permanova_jaccard_diet_pairwise.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "cc9a7144-8261-4490-a77a-de8be425084f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/856e2168-5ce8-46ca-a4c3-4d358cd172c6')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 856e2168-5ce8-46ca-a4c3-4d358cd172c6>"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/permanova/permanova_bray_diet_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "9d5fce74-e7d3-4b0f-8497-06d3af0094de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/6babe64c-5724-4e72-af9d-91ff3d927261')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 6babe64c-5724-4e72-af9d-91ff3d927261>"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/permanova/permanova_jaccard_diet_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82d38993-cffe-4331-8701-e94e7a22c14d",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### IDB Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "632e1cc7-69aa-4d1a-a625-095810950416",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/permanova/permanova_bray_ibd_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/06/permanova/permanova_jaccard_ibd_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Pairwise Bray-Curtis\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/06/boots-kmer-diversity/distance_matrices/braycurtis.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column ibd_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/06/permanova/permanova_bray_ibd_pairwise.qzv\n",
    "\n",
    "# Pairwise Jaccard\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/06/boots-kmer-diversity/distance_matrices/jaccard.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column ibd_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/06/permanova/permanova_jaccard_ibd_pairwise.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "204a0113-b5ec-46b1-883c-3926371c07d3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/e0e47c2b-ea90-41ec-8d26-c48da9586e79')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e0e47c2b-ea90-41ec-8d26-c48da9586e79>"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/permanova/permanova_bray_ibd_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "654c61c8-27d8-4d88-b68e-1a5c68a34bfd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/dcdb8bd8-8ecf-44d8-99f1-b9cf49e7d75a')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: dcdb8bd8-8ecf-44d8-99f1-b9cf49e7d75a>"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/06/permanova/permanova_jaccard_ibd_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0a05745-0c32-4de0-bc50-6efe3c692150",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### BMI "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a8c81e7-be4d-400e-8799-f5de7d705a30",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45f6c3a1-6d74-43f7-806e-3304603fd22f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e47af34e-75ce-4fa8-a5ce-97dcefc17b54",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "b3e00c54-d43c-4ea2-a5c6-f5412ec10a55",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Celiac disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ceb6eeb-9bcc-4592-9e3f-bd80b016ec46",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e208864-9450-40b6-a757-c41d4e44a7c3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa916cb6-e08c-4975-b6d5-141e1fde7675",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "496079b0-3f4f-44ae-a6f8-b0498976cf25",
   "metadata": {},
   "source": [
    "XXXXXXXXXXXXXXX\n",
    "\n",
    "Columns for PERMANOVA:\n",
    "country_sample\n",
    "sex_sample\n",
    "diet_type_sample\n",
    "BMI_category\n",
    "urban/rural/peri\n",
    "\n",
    "XXXXXXXXXXXXXX"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7517fa41-577e-4c17-8315-c63af9099fe2",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 07 Differential Abundance üßÆ"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "23e2e0be-2ebd-432d-9960-636c010c8357",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 07.01 Filtering our features"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d7f6bd3-9930-4dc6-9792-5aef66ffd069",
   "metadata": {
    "tags": []
   },
   "source": [
    "Before we start with the *actual\" abundnace analysis, let's prepare for it by improving the resolution of the downstream analysis. We do so by first filtering our feature table to only keep features that appear at least 25 times and in at least 4 samples. Then, we will collapse our taxonomic classification at the species level (=7) so that we can later check and compare the differential abundance both at the ASV level and at the species level."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "c6faf937-026c-459e-8eb4-1474a6368795",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/test_table_abundance_before_filtering.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# FT before filtering\n",
    "# Summarize feature table (check per-sample read depth, number of features, and overall counts)\n",
    "! qiime feature-table summarize \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "    --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/07/test_table_abundance_before_filtering.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "261f5719-3a47-4797-b0ae-f8ffa1b50803",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/d600e9f2-5bbe-45b6-9c9d-87b16dc070e5')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: d600e9f2-5bbe-45b6-9c9d-87b16dc070e5>"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/test_table_abundance_before_filtering.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "39282e53-ff27-40c4-8d10-640c1f1d61f4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/07/table_abundance.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime feature-table filter-features \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "    --p-min-frequency 25 \\\n",
    "    --p-min-samples 4 \\\n",
    "    --o-filtered-table $data_dir/07/table_abundance.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "a2066801-2e19-454c-b90b-c9e3226d5ff6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/07/table_abundance_20_3.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime feature-table filter-features \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "    --p-min-frequency 20 \\\n",
    "    --p-min-samples 3 \\\n",
    "    --o-filtered-table $data_dir/07/table_abundance_20_3.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "29731a00-a990-48a1-9c1a-fe2b9cc65664",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/table_abundance.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Summarize feature table (check per-sample read depth, number of features, and overall counts)\n",
    "! qiime feature-table summarize \\\n",
    "    --i-table $data_dir/07/table_abundance.qza \\\n",
    "    --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/07/table_abundance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "de8cd673-9e7d-42aa-81d6-e7d810cc826b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/table_abundance_20_3.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Summarize feature table (check per-sample read depth, number of features, and overall counts)\n",
    "! qiime feature-table summarize \\\n",
    "    --i-table $data_dir/07/table_abundance_20_3.qza \\\n",
    "    --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/07/table_abundance_20_3.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "23858d22-5163-4288-9be3-051cb7f0696b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/dc4e4f2d-6099-45f4-9e31-6c60c6e629b1')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: dc4e4f2d-6099-45f4-9e31-6c60c6e629b1>"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/table_abundance_20_3.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "a66db339-01c6-4b97-b167-54ec1183f66a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/1f7bf0f6-42e0-4420-82d6-23f8e2fa7e78')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 1f7bf0f6-42e0-4420-82d6-23f8e2fa7e78>"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/table_abundance.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "ffc852f8-cc87-4bd1-9646-7db75e08d23c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/07/table_abundance_collapsed.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "! qiime taxa collapse \\\n",
    "    --i-table $data_dir/07/table_abundance.qza \\\n",
    "    --i-taxonomy $data_dir/04/taxjob_euler.qza \\\n",
    "    --p-level 7 \\\n",
    "    --o-collapsed-table $data_dir/07/table_abundance_collapsed.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "5fd66111-a8bc-4610-a651-0a12752aec07",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/table_abundance_collapsed.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Summarize feature table (check per-sample read depth, number of features, and overall counts)\n",
    "! qiime feature-table summarize \\\n",
    "    --i-table $data_dir/07/table_abundance_collapsed.qza \\\n",
    "    --m-sample-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --o-visualization $data_dir/07/table_abundance_collapsed.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "a0272e93-a550-4398-aabf-dc74ba775f4b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/048d2b6e-da48-49d8-9fba-75d683b3b2fa')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 048d2b6e-da48-49d8-9fba-75d683b3b2fa>"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/table_abundance_collapsed.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "886889ac-e9ed-4fb2-9c22-8a84b1173dec",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 07.02 Ancom-BC"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc0f1497-9dc8-47e0-b1a7-06b56d45e1d8",
   "metadata": {},
   "source": [
    "Great! Let's now run the actual differnetial analysis, we will run Aconm-BC on the metadata variable of interest on both the ASV level and the species level."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78aa3a99-32cb-4533-a028-89feda54bfd1",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Diet type"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9c724d5-94e0-4c88-99a0-a0efe546a3fd",
   "metadata": {
    "tags": []
   },
   "source": [
    "First for ASV level then also on species level (level 7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "f8794998-905b-4ced-8a27-fb3927696066",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureData[DifferentialAbundance] to: data/07/ancombc/ancombc_diet_differentials.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/ancombc/ancombc_diet_da_barplot.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/ancombc/ancombc_diet_results.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "#ASV LEVEL:\n",
    "# Run ANCOM-BC\n",
    "! qiime composition ancombc \\\n",
    "    --i-table $data_dir/07/table_abundance.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --p-formula diet_type_sample \\\n",
    "    --o-differentials $data_dir/07/ancombc/ancombc_diet_differentials.qza\n",
    "\n",
    "# Generate a barplot of differentially abundant taxa between environments\n",
    "! qiime composition da-barplot \\\n",
    "    --i-data $data_dir/07/ancombc/ancombc_diet_differentials.qza \\\n",
    "    --o-visualization $data_dir/07/ancombc/ancombc_diet_da_barplot.qzv\n",
    "\n",
    "# Generate a table of these same values for all taxa\n",
    "! qiime composition tabulate \\\n",
    "    --i-data $data_dir/07/ancombc/ancombc_diet_differentials.qza \\\n",
    "    --o-visualization $data_dir/07/ancombc/ancombc_diet_results.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "3a30b54d-e459-4fdb-a81d-751d8eab273c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureData[DifferentialAbundance] to: data/07/ancombc/ancombc_diet_differentials_species.qza\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/ancombc/ancombc_diet_da_barplot_species.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/07/ancombc/ancombc_diet_results_species.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "#Species LEVEL:\n",
    "# Run ANCOM-BC\n",
    "! qiime composition ancombc \\\n",
    "    --i-table $data_dir/07/table_abundance_collapsed.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --p-formula diet_type_sample \\\n",
    "    --o-differentials $data_dir/07/ancombc/ancombc_diet_differentials_species.qza\n",
    "\n",
    "# Generate a barplot of differentially abundant taxa between environments\n",
    "! qiime composition da-barplot \\\n",
    "    --i-data $data_dir/07/ancombc/ancombc_diet_differentials_species.qza \\\n",
    "    --o-visualization $data_dir/07/ancombc/ancombc_diet_da_barplot_species.qzv\n",
    "\n",
    "# Generate a table of these same values for all taxa\n",
    "! qiime composition tabulate \\\n",
    "    --i-data $data_dir/07/ancombc/ancombc_diet_differentials_species.qza \\\n",
    "    --o-visualization $data_dir/07/ancombc/ancombc_diet_results_species.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71a6e4c3-5f4a-46d9-ad54-51e7130d6edf",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### ASV Level:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "6ec467f5-7b09-49e4-8f3c-6b54b38d94aa",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/6f59427b-1654-4f76-84f3-9875b940033f')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 6f59427b-1654-4f76-84f3-9875b940033f>"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/ancombc/ancombc_diet_da_barplot.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "9cbb0c98-9540-4419-94d1-790f1ef0d839",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/97f1e647-e060-4f1e-8430-ddbf8fa579be')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 97f1e647-e060-4f1e-8430-ddbf8fa579be>"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/ancombc/ancombc_diet_results.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fed8938e-c82d-4e47-bc65-90e1cca88acb",
   "metadata": {
    "tags": []
   },
   "source": [
    "##### Species Level:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "e2fbef7b-d017-4a2f-ae77-3d9476f7c452",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/383c78f1-a8e2-453b-bf57-90dfb0badbbc')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 383c78f1-a8e2-453b-bf57-90dfb0badbbc>"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/ancombc/ancombc_diet_da_barplot_species.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "45a61c03-580a-466b-95f3-51ab269a833d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/afbbef8d-ea49-4a86-892e-81dca47d50db')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: afbbef8d-ea49-4a86-892e-81dca47d50db>"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/07/ancombc/ancombc_diet_results_species.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d9227bb-27d6-4da6-8371-2900d356e477",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# 08 Diversity analysis and differential Abundance WITHOUT using Bootstrapping ü•æüö´"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f45e238-fb3a-4644-a339-b0711349860d",
   "metadata": {},
   "source": [
    "Having performed both diversity analysis and differnetial abundance anylsis on our bootstrapped and kmerized sequences, we now want to do both analyses withouth bootstrapping to accurately compare the two pipelines."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a90ff2f7-d9c7-4bef-8853-000637e8e49d",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 08.01 Making the core metrics (kmerizer)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "11fbfca6-3ede-4d73-8a3f-cbf68d6e4229",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/08/kmerizer-results/rarefied_table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: data/08/kmerizer-results/kmer_table.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: data/08/kmerizer-results/observed_features_vector.qza\u001b[0m\n",
      "\u001b[32mSaved SampleData[AlphaDiversity] to: data/08/kmerizer-results/shannon_vector.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: data/08/kmerizer-results/jaccard_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved DistanceMatrix to: data/08/kmerizer-results/bray_curtis_distance_matrix.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: data/08/kmerizer-results/jaccard_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved PCoAResults to: data/08/kmerizer-results/bray_curtis_pcoa_results.qza\u001b[0m\n",
      "\u001b[32mSaved Visualization to: data/08/kmerizer-results/scatterplot.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime kmerizer core-metrics \\\n",
    "    --i-table $data_dir/04/dada2_table_filtered_samples_removed.qza \\\n",
    "    --i-sequences $data_dir/04/dada2_rep_seqs_filtered.qza \\\n",
    "    --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "    --p-sampling-depth 20000 \\\n",
    "    --p-kmer-size 16 \\\n",
    "    --output-dir $data_dir/08/kmerizer-results"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e1c0fc2-a8fe-42a5-b413-111b7ee77be6",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## 08.02 Alpha diversity (kmerizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a57a0ec-d084-4685-b2a3-a3db3054c501",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Categorical Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "37c89d09-8dc4-4ad0-8ed3-bfb15ad5b58d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/alpha_diversity/alpha_shannon_group_significance_.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/alpha_diversity/alpha_observed_features_group_significance.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity $data_dir/08/kmerizer-results/shannon_vector.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/alpha_diversity/alpha_shannon_group_significance_.qzv\n",
    "\n",
    "!qiime diversity alpha-group-significance \\\n",
    "  --i-alpha-diversity $data_dir/08/kmerizer-results/observed_features_vector.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/alpha_diversity/alpha_observed_features_group_significance.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1d2c7337-fdae-4b94-aaac-b60bc233ad14",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/a2731e06-7e64-4dbc-928d-568e886541d7')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: a2731e06-7e64-4dbc-928d-568e886541d7>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/alpha_diversity/alpha_shannon_group_significance_.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "81ea1a8b-e1b0-48ed-a61a-9c9a66e07ae7",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/947f4e63-f89e-4b85-b700-78bb58a234bc')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 947f4e63-f89e-4b85-b700-78bb58a234bc>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/alpha_diversity/alpha_observed_features_group_significance.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b451d6d-346d-42c2-a5f6-71bf3569c978",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Numeric Variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "bf080730-c44a-48eb-80fa-d076c3b6229c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/alpha_diversity/alpha_shannon_corrleation.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/alpha_diversity/alpha_observed_features_correlation.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "!qiime diversity alpha-correlation \\\n",
    "  --i-alpha-diversity $data_dir/08/kmerizer-results/shannon_vector.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/alpha_diversity/alpha_shannon_corrleation.qzv\n",
    "\n",
    "!qiime diversity alpha-correlation \\\n",
    "  --i-alpha-diversity $data_dir/08/kmerizer-results/observed_features_vector.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/alpha_diversity/alpha_observed_features_correlation.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "18e4db90-a269-456e-89c6-25f3f955018a",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/4dee788e-a3db-4121-aef7-e02effba5d92')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4dee788e-a3db-4121-aef7-e02effba5d92>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/alpha_diversity/alpha_shannon_corrleation.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d43ed015-9059-4562-83d6-61a513a4fd27",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/5ea48a0e-78b6-4421-9486-2f8cba47a684')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 5ea48a0e-78b6-4421-9486-2f8cba47a684>"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/alpha_diversity/alpha_observed_features_correlation.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d6eefd1-04a5-4dc8-84da-4018508efa4c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### 06.02.2 Alpha diversity findings"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "331fd0ed-cd9c-4cdc-a86b-a2bbe0dce23b",
   "metadata": {
    "tags": []
   },
   "source": [
    "<div style=\"background-color: yellow; padding: 20px;\">\n",
    "Analysing our alpha diversity plots for all three metrics of shannon, observed features and pielou we found the following things:\n",
    "</div>\n",
    "\n",
    "**Numeric variables:**\n",
    "- Age_years_sample is relevant only for richnenss (pValue=0.0174 in observed_feature alpha correlation)\n",
    "- All other numeric variables showed no significance for all three metrics\n",
    "\n",
    "**Categorical Variables:**\n",
    "- Statistical differnce found for **diet type**: both shannon and observed features show relevant p-Values (Shannon: 0.0273, observed features: 0.0274) for comaprisons to all other diet types (Vegan sample size n=5)\n",
    "- Statistical differnce found for **IDB status**: Significant p-Value (0.0119) for shannon metric when comparing participants with diagnosed IDB (n=10) to participants without (n=121).\n",
    "\n",
    "- No statistically relevant differences for the following variables:\n",
    "    - Country\n",
    "    - State\n",
    "    - Sex\n",
    "    - Celiac disease\n",
    "    - BMI\n",
    "    - Rural/Urban/peri"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f9bdfbc-d04b-4cf8-90fc-b3f9725309ab",
   "metadata": {
    "tags": []
   },
   "source": [
    "## 08.03 Beta diversity (kmerizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcc8e27f-448d-44ae-9333-3408ab5f7646",
   "metadata": {},
   "source": [
    "To investigate between-sample diversity (=beta diversity) we take the beta diversity outputs from the kmerizer directory and convert them to emperor plots. In the plots we can then search for any clustering of our samples (spoiler alert: we don't find any clear clustering here either)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "247f6288-88c4-4bdf-baec-8aedba7cd4d1",
   "metadata": {
    "tags": []
   },
   "source": [
    "### 08.03.01 Making the plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c9d77dc5-55af-438e-b77f-20d041ea13e5",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/beta_diversity_plots/kmer_bray_emperor.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/beta_diversity_plots/kmer_jaccard_emperor.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Bray‚ÄìCurtis\n",
    "!qiime emperor plot \\\n",
    "  --i-pcoa $data_dir/08/kmerizer-results/bray_curtis_pcoa_results.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/beta_diversity_plots/kmer_bray_emperor.qzv\n",
    "\n",
    "# Jaccard\n",
    "!qiime emperor plot \\\n",
    "  --i-pcoa $data_dir/08/kmerizer-results/jaccard_pcoa_results.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --o-visualization $data_dir/08/beta_diversity_plots/kmer_jaccard_emperor.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "b442f3fd-d8c8-4341-8220-14f3f63073bb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/4c3192c0-db35-48d2-870b-6658a2b740f3')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 4c3192c0-db35-48d2-870b-6658a2b740f3>"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/beta_diversity_plots/kmer_bray_emperor.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1f7179db-8423-4403-8312-b5c29a326424",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/f7d6469e-4ce0-4613-a78e-d63ac0cb9e35')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: f7d6469e-4ce0-4613-a78e-d63ac0cb9e35>"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/beta_diversity_plots/kmer_jaccard_emperor.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "011775f7-0cd1-49da-b67b-9cd908eec4ca",
   "metadata": {
    "tags": []
   },
   "source": [
    "### 08.03.02 PERMANOVA (kmerizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "296404ea-60e7-449e-8514-8851a41ddc87",
   "metadata": {},
   "source": [
    "XXXX Having found no clear clustering in our beta diversity plots, let us now check if we find statistically relevant differences between our samples using PERMANOVA. \n",
    "\n",
    "Variables of interest:\n",
    "- Diet type\n",
    "- IDB Status\n",
    "- BMI"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4bef0d1c-13d1-4764-8e81-2a5882a24418",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Diet type"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a7493ef2-5ba3-4e29-8433-2dc470f4b234",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/permanova/permanova_bray_diet_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/permanova/permanova_jaccard_diet_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Pairwise Bray-Curtis\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/08/kmerizer-results/bray_curtis_distance_matrix.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column diet_type_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/08/permanova/permanova_bray_diet_pairwise.qzv\n",
    "\n",
    "# Pairwise Jaccard\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/08/kmerizer-results/jaccard_distance_matrix.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column diet_type_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/08/permanova/permanova_jaccard_diet_pairwise.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "79aa0dda-6b9f-49ed-aa06-e641372a5119",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/d70039d2-18e1-4ba7-a01e-ceb59cfdae33')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: d70039d2-18e1-4ba7-a01e-ceb59cfdae33>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/permanova/permanova_bray_diet_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "bd98e6f9-a8be-492b-af3d-84c5f67d9edc",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/27153ff5-0fbf-4ffa-b035-7a7fdbc7f4e5')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: 27153ff5-0fbf-4ffa-b035-7a7fdbc7f4e5>"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/permanova/permanova_jaccard_diet_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80d95ef6-b92a-4a0b-b157-caffc78da955",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### IDB Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b92b60ab-255b-45c7-afd7-4a267dcef9c6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/permanova/permanova_bray_ibd_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h/opt/conda/lib/python3.10/site-packages/unifrac/__init__.py:9: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n",
      "\u001b[32mSaved Visualization to: data/08/permanova/permanova_jaccard_ibd_pairwise.qzv\u001b[0m\n",
      "\u001b[0m\u001b[?25h"
     ]
    }
   ],
   "source": [
    "# Pairwise Bray-Curtis\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/08/kmerizer-results/bray_curtis_distance_matrix.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column ibd_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/08/permanova/permanova_bray_ibd_pairwise.qzv\n",
    "\n",
    "# Pairwise Jaccard\n",
    "!qiime diversity beta-group-significance \\\n",
    "  --i-distance-matrix $data_dir/08/kmerizer-results/jaccard_distance_matrix.qza \\\n",
    "  --m-metadata-file $data_dir/metadata_ed_rem.tsv \\\n",
    "  --m-metadata-column ibd_sample \\\n",
    "  --p-method permanova \\\n",
    "  --p-permutations 999 \\\n",
    "  --p-pairwise \\\n",
    "  --o-visualization $data_dir/08/permanova/permanova_jaccard_ibd_pairwise.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "b022e2df-f171-440c-8251-f3bbfb211ef9",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/c862a4cb-a838-44f5-8911-7b427573846d')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: c862a4cb-a838-44f5-8911-7b427573846d>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/permanova/permanova_bray_ibd_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "f1d99ee7-b46a-4562-9d17-17be4f4f01cd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><img onload=\"(function(div, url){\n",
       "if (typeof require !== 'undefined') {\n",
       "    var baseURL = require.toUrl('').split('/').slice(0, -2).join('/');\n",
       "} else {\n",
       "    var baseURL = JSON.parse(\n",
       "        document.getElementById('jupyter-config-data').innerHTML\n",
       "    ).baseUrl.slice(0, -1);\n",
       "}\n",
       "url = baseURL + url;\n",
       "fetch(url).then(function(res) {\n",
       "    if (res.status === 404) {\n",
       "        div.innerHTML = 'Install QIIME 2 Jupyter extension with:<br />' +\n",
       "                        '<code>jupyter server extension enable --py qiime2' +\n",
       "                        ' --sys-prefix</code><br />then restart your server.' +\n",
       "                        '<br /><br />(Interactive output not available on ' +\n",
       "                        'static notebook viewer services like nbviewer.)';\n",
       "    } else if (res.status === 409) {\n",
       "        div.innerHTML = 'Visualization no longer in scope. Re-run this cell' +\n",
       "                        ' to see the visualization.';\n",
       "    } else if (res.ok) {\n",
       "        url = res.url;\n",
       "        div.innerHTML = '<iframe src=\\'' + url + '\\' style=\\'' +\n",
       "                        'width: 100%; height: 700px; border: 0;\\'>' +\n",
       "                        '</iframe><hr />Open in a: <a href=\\'' + url + '\\'' +\n",
       "                        ' target=\\'_blank\\'>new window</a>'\n",
       "    } else {\n",
       "        div.innerHTML = 'Something has gone wrong. Check notebook server for' +\n",
       "                        ' errors.';\n",
       "    }\n",
       "});\n",
       "})(this.parentElement, '/qiime2/redirect?location=/tmp/qiime2/jovyan/data/e7331e2f-c83e-495c-a9e9-13b0701528e6')\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==\" /></div>"
      ],
      "text/plain": [
       "<visualization: Visualization uuid: e7331e2f-c83e-495c-a9e9-13b0701528e6>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Visualization.load(f\"{data_dir}/08/permanova/permanova_jaccard_ibd_pairwise.qzv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a539ffb-dd6f-4fe7-9d4e-a04cda9fcc5f",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### BMI "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebc805c2-6a43-4636-a2e4-c17ace415a4c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5e6f1f16-5f1e-4d5a-a74e-2fdaa08e1b9c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88b694d4-9819-4bf1-bbd7-243408a8fc87",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "692475ba-8db3-43d1-b24d-8100cf89f2ad",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Celiac disease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c3fc94f-7c62-463d-bb0d-8b483026e669",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf0827a8-9280-4296-873a-e199d3290148",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26a9d123-79d7-4b83-a1b9-f62831acb77e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "QIIME 2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
